{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# InterMineR Workshop Use Case\n",
    "\n",
    "\n",
    "\n",
    "We are going to re-create the workflow we did using the web interface using the R API.\n",
    "\n",
    "The basic steps are:  \n",
    "\n",
    "1. Load the InterMine library and choose an InterMine to query.\n",
    "2. Query 1: Diabetes Genes: Fetch a list of genes that are associated with diabetes\n",
    "3. Query 2: PAX6 + Pancreas: Fetch a list of genes that have medium or high expression in the pancreas and are in our PAX6 targets list\n",
    "4. Intersection: Find which genes are present in _both_ Query 1 and Query2.\n",
    "5. GWAS: Compare the intersection of the previous query with results from GWAS studies."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting started - Load InterMineR and choose an InterMine"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the InterMine library. If it's not already installed, visit\n",
    "https://bioconductor.org/packages/release/bioc/html/InterMineR.html and follow the instructions to install.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(InterMineR)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We want to query human data - so let's look and see what InterMines are available: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>BMAP</dt>\n",
       "\t\t<dd>'https://bmap.jgi.doe.gov/bmapmine/'</dd>\n",
       "\t<dt>BeanMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/beanmine'</dd>\n",
       "\t<dt>BovineMine</dt>\n",
       "\t\t<dd>'http://genomes.missouri.edu/bovinemine'</dd>\n",
       "\t<dt>CHOmine</dt>\n",
       "\t\t<dd>'https://chomine.boku.ac.at/chomine'</dd>\n",
       "\t<dt>ChickpeaMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/chickpeamine'</dd>\n",
       "\t<dt>CowpeaMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/cowpeamine'</dd>\n",
       "\t<dt>FlyMine</dt>\n",
       "\t\t<dd>'http://www.flymine.org/flymine'</dd>\n",
       "\t<dt>GrapeMine</dt>\n",
       "\t\t<dd>'http://urgi.versailles.inra.fr/GrapeMine'</dd>\n",
       "\t<dt>HumanMine</dt>\n",
       "\t\t<dd>'http://www.humanmine.org/humanmine'</dd>\n",
       "\t<dt>HymenopteraMine</dt>\n",
       "\t\t<dd>'http://hymenopteragenome.org/hymenopteramine'</dd>\n",
       "\t<dt>IndigoMine</dt>\n",
       "\t\t<dd>'http://www.cbrc.kaust.edu.sa/indigo'</dd>\n",
       "\t<dt>LegumeMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/legumemine'</dd>\n",
       "\t<dt>MaizeMine</dt>\n",
       "\t\t<dd>'http://maizemine.rnet.missouri.edu:8080/maizemine'</dd>\n",
       "\t<dt>MedicMine</dt>\n",
       "\t\t<dd>'http://medicmine.jcvi.org/medicmine'</dd>\n",
       "\t<dt>MitoMiner</dt>\n",
       "\t\t<dd>'http://mitominer.mrc-mbu.cam.ac.uk/release-4.0'</dd>\n",
       "\t<dt>ModMine</dt>\n",
       "\t\t<dd>'http://intermine.modencode.org/release-33'</dd>\n",
       "\t<dt>MouseMine</dt>\n",
       "\t\t<dd>'http://www.mousemine.org/mousemine'</dd>\n",
       "\t<dt>OakMine</dt>\n",
       "\t\t<dd>'https://urgi.versailles.inra.fr/OakMine_PM1N'</dd>\n",
       "\t<dt>PeanutMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/peanutmine'</dd>\n",
       "\t<dt>PhytoMine</dt>\n",
       "\t\t<dd>'https://phytozome.jgi.doe.gov/phytomine/'</dd>\n",
       "\t<dt>PlanMine</dt>\n",
       "\t\t<dd>'http://planmine.mpi-cbg.de/planmine'</dd>\n",
       "\t<dt>RatMine</dt>\n",
       "\t\t<dd>'http://ratmine.mcw.edu/ratmine'</dd>\n",
       "\t<dt>RepetDB</dt>\n",
       "\t\t<dd>'http://urgi.versailles.inra.fr/repetdb'</dd>\n",
       "\t<dt>SoyMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/soymine'</dd>\n",
       "\t<dt>TargetMine</dt>\n",
       "\t\t<dd>'https://targetmine.mizuguchilab.org/targetmine'</dd>\n",
       "\t<dt>ThaleMine</dt>\n",
       "\t\t<dd>'https://apps.araport.org/thalemine'</dd>\n",
       "\t<dt>WheatMine</dt>\n",
       "\t\t<dd>'https://urgi.versailles.inra.fr/WheatMine'</dd>\n",
       "\t<dt>WormMine</dt>\n",
       "\t\t<dd>'http://intermine.wormbase.org/tools/wormmine/'</dd>\n",
       "\t<dt>XenMine</dt>\n",
       "\t\t<dd>'http://www.xenmine.org/xenmine'</dd>\n",
       "\t<dt>YeastMine</dt>\n",
       "\t\t<dd>'https://yeastmine.yeastgenome.org/yeastmine'</dd>\n",
       "\t<dt>ZebrafishMine</dt>\n",
       "\t\t<dd>'http://zebrafishmine.org'</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[BMAP] 'https://bmap.jgi.doe.gov/bmapmine/'\n",
       "\\item[BeanMine] 'https://mines.legumeinfo.org/beanmine'\n",
       "\\item[BovineMine] 'http://genomes.missouri.edu/bovinemine'\n",
       "\\item[CHOmine] 'https://chomine.boku.ac.at/chomine'\n",
       "\\item[ChickpeaMine] 'https://mines.legumeinfo.org/chickpeamine'\n",
       "\\item[CowpeaMine] 'https://mines.legumeinfo.org/cowpeamine'\n",
       "\\item[FlyMine] 'http://www.flymine.org/flymine'\n",
       "\\item[GrapeMine] 'http://urgi.versailles.inra.fr/GrapeMine'\n",
       "\\item[HumanMine] 'http://www.humanmine.org/humanmine'\n",
       "\\item[HymenopteraMine] 'http://hymenopteragenome.org/hymenopteramine'\n",
       "\\item[IndigoMine] 'http://www.cbrc.kaust.edu.sa/indigo'\n",
       "\\item[LegumeMine] 'https://mines.legumeinfo.org/legumemine'\n",
       "\\item[MaizeMine] 'http://maizemine.rnet.missouri.edu:8080/maizemine'\n",
       "\\item[MedicMine] 'http://medicmine.jcvi.org/medicmine'\n",
       "\\item[MitoMiner] 'http://mitominer.mrc-mbu.cam.ac.uk/release-4.0'\n",
       "\\item[ModMine] 'http://intermine.modencode.org/release-33'\n",
       "\\item[MouseMine] 'http://www.mousemine.org/mousemine'\n",
       "\\item[OakMine] 'https://urgi.versailles.inra.fr/OakMine\\_PM1N'\n",
       "\\item[PeanutMine] 'https://mines.legumeinfo.org/peanutmine'\n",
       "\\item[PhytoMine] 'https://phytozome.jgi.doe.gov/phytomine/'\n",
       "\\item[PlanMine] 'http://planmine.mpi-cbg.de/planmine'\n",
       "\\item[RatMine] 'http://ratmine.mcw.edu/ratmine'\n",
       "\\item[RepetDB] 'http://urgi.versailles.inra.fr/repetdb'\n",
       "\\item[SoyMine] 'https://mines.legumeinfo.org/soymine'\n",
       "\\item[TargetMine] 'https://targetmine.mizuguchilab.org/targetmine'\n",
       "\\item[ThaleMine] 'https://apps.araport.org/thalemine'\n",
       "\\item[WheatMine] 'https://urgi.versailles.inra.fr/WheatMine'\n",
       "\\item[WormMine] 'http://intermine.wormbase.org/tools/wormmine/'\n",
       "\\item[XenMine] 'http://www.xenmine.org/xenmine'\n",
       "\\item[YeastMine] 'https://yeastmine.yeastgenome.org/yeastmine'\n",
       "\\item[ZebrafishMine] 'http://zebrafishmine.org'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "BMAP\n",
       ":   'https://bmap.jgi.doe.gov/bmapmine/'BeanMine\n",
       ":   'https://mines.legumeinfo.org/beanmine'BovineMine\n",
       ":   'http://genomes.missouri.edu/bovinemine'CHOmine\n",
       ":   'https://chomine.boku.ac.at/chomine'ChickpeaMine\n",
       ":   'https://mines.legumeinfo.org/chickpeamine'CowpeaMine\n",
       ":   'https://mines.legumeinfo.org/cowpeamine'FlyMine\n",
       ":   'http://www.flymine.org/flymine'GrapeMine\n",
       ":   'http://urgi.versailles.inra.fr/GrapeMine'HumanMine\n",
       ":   'http://www.humanmine.org/humanmine'HymenopteraMine\n",
       ":   'http://hymenopteragenome.org/hymenopteramine'IndigoMine\n",
       ":   'http://www.cbrc.kaust.edu.sa/indigo'LegumeMine\n",
       ":   'https://mines.legumeinfo.org/legumemine'MaizeMine\n",
       ":   'http://maizemine.rnet.missouri.edu:8080/maizemine'MedicMine\n",
       ":   'http://medicmine.jcvi.org/medicmine'MitoMiner\n",
       ":   'http://mitominer.mrc-mbu.cam.ac.uk/release-4.0'ModMine\n",
       ":   'http://intermine.modencode.org/release-33'MouseMine\n",
       ":   'http://www.mousemine.org/mousemine'OakMine\n",
       ":   'https://urgi.versailles.inra.fr/OakMine_PM1N'PeanutMine\n",
       ":   'https://mines.legumeinfo.org/peanutmine'PhytoMine\n",
       ":   'https://phytozome.jgi.doe.gov/phytomine/'PlanMine\n",
       ":   'http://planmine.mpi-cbg.de/planmine'RatMine\n",
       ":   'http://ratmine.mcw.edu/ratmine'RepetDB\n",
       ":   'http://urgi.versailles.inra.fr/repetdb'SoyMine\n",
       ":   'https://mines.legumeinfo.org/soymine'TargetMine\n",
       ":   'https://targetmine.mizuguchilab.org/targetmine'ThaleMine\n",
       ":   'https://apps.araport.org/thalemine'WheatMine\n",
       ":   'https://urgi.versailles.inra.fr/WheatMine'WormMine\n",
       ":   'http://intermine.wormbase.org/tools/wormmine/'XenMine\n",
       ":   'http://www.xenmine.org/xenmine'YeastMine\n",
       ":   'https://yeastmine.yeastgenome.org/yeastmine'ZebrafishMine\n",
       ":   'http://zebrafishmine.org'\n",
       "\n"
      ],
      "text/plain": [
       "                                               BMAP \n",
       "               \"https://bmap.jgi.doe.gov/bmapmine/\" \n",
       "                                           BeanMine \n",
       "            \"https://mines.legumeinfo.org/beanmine\" \n",
       "                                         BovineMine \n",
       "           \"http://genomes.missouri.edu/bovinemine\" \n",
       "                                            CHOmine \n",
       "               \"https://chomine.boku.ac.at/chomine\" \n",
       "                                       ChickpeaMine \n",
       "        \"https://mines.legumeinfo.org/chickpeamine\" \n",
       "                                         CowpeaMine \n",
       "          \"https://mines.legumeinfo.org/cowpeamine\" \n",
       "                                            FlyMine \n",
       "                   \"http://www.flymine.org/flymine\" \n",
       "                                          GrapeMine \n",
       "         \"http://urgi.versailles.inra.fr/GrapeMine\" \n",
       "                                          HumanMine \n",
       "               \"http://www.humanmine.org/humanmine\" \n",
       "                                    HymenopteraMine \n",
       "     \"http://hymenopteragenome.org/hymenopteramine\" \n",
       "                                         IndigoMine \n",
       "              \"http://www.cbrc.kaust.edu.sa/indigo\" \n",
       "                                         LegumeMine \n",
       "          \"https://mines.legumeinfo.org/legumemine\" \n",
       "                                          MaizeMine \n",
       "\"http://maizemine.rnet.missouri.edu:8080/maizemine\" \n",
       "                                          MedicMine \n",
       "              \"http://medicmine.jcvi.org/medicmine\" \n",
       "                                          MitoMiner \n",
       "   \"http://mitominer.mrc-mbu.cam.ac.uk/release-4.0\" \n",
       "                                            ModMine \n",
       "        \"http://intermine.modencode.org/release-33\" \n",
       "                                          MouseMine \n",
       "               \"http://www.mousemine.org/mousemine\" \n",
       "                                            OakMine \n",
       "     \"https://urgi.versailles.inra.fr/OakMine_PM1N\" \n",
       "                                         PeanutMine \n",
       "          \"https://mines.legumeinfo.org/peanutmine\" \n",
       "                                          PhytoMine \n",
       "         \"https://phytozome.jgi.doe.gov/phytomine/\" \n",
       "                                           PlanMine \n",
       "              \"http://planmine.mpi-cbg.de/planmine\" \n",
       "                                            RatMine \n",
       "                   \"http://ratmine.mcw.edu/ratmine\" \n",
       "                                            RepetDB \n",
       "           \"http://urgi.versailles.inra.fr/repetdb\" \n",
       "                                            SoyMine \n",
       "             \"https://mines.legumeinfo.org/soymine\" \n",
       "                                         TargetMine \n",
       "   \"https://targetmine.mizuguchilab.org/targetmine\" \n",
       "                                          ThaleMine \n",
       "               \"https://apps.araport.org/thalemine\" \n",
       "                                          WheatMine \n",
       "        \"https://urgi.versailles.inra.fr/WheatMine\" \n",
       "                                           WormMine \n",
       "    \"http://intermine.wormbase.org/tools/wormmine/\" \n",
       "                                            XenMine \n",
       "                   \"http://www.xenmine.org/xenmine\" \n",
       "                                          YeastMine \n",
       "      \"https://yeastmine.yeastgenome.org/yeastmine\" \n",
       "                                      ZebrafishMine \n",
       "                         \"http://zebrafishmine.org\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "listMines()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Okay, let's select HumanMine from the list:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>HumanMine:</strong> 'http://www.humanmine.org/humanmine'"
      ],
      "text/latex": [
       "\\textbf{HumanMine:} 'http://www.humanmine.org/humanmine'"
      ],
      "text/markdown": [
       "**HumanMine:** 'http://www.humanmine.org/humanmine'"
      ],
      "text/plain": [
       "                           HumanMine \n",
       "\"http://www.humanmine.org/humanmine\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "humanMine <- listMines()[\"HumanMine\"] #select humanmine\n",
    "humanMine #print out the value to see what's inside"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Okay, now let's tell InterMineR that we want to use HumanMine for our queries.\n",
    "\n",
    "**Important:** you'll need an API token for this part so you can access your HumanMine account. You can get your token by logging into [HumanMine](http://www.humanmine.org/) and going to the account details tab within MyMine. Cut and paste your token into the code below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "im <- initInterMine(mine=humanMine, \"YOUR TOKEN HERE\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### First Query: Diabetes Genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Our first query will be to select all human genes that are associate with diabetes. This will require two constraints:\n",
    "\n",
    "1. Ensure all genes returned are `Home sapiens` genes (HumanMine contains some non-human genes for homology query purposes)\n",
    "2. Restrict results to genes that are associated with `diabetes`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "query1Diabetes <- setQuery( \n",
    "  # here we're choosing which columns of data we'd like to see\n",
    "  select = c(\"Gene.primaryIdentifier\", \"Gene.symbol\"),\n",
    "    # set the logic for constraints. The first constraint is the first path+operator+value, \n",
    "    # e.g. Gene.organism.name = Homo sapiens, and the second constraint is the combination \n",
    "    # of the second path+operator+value, e.g. Gene.diseases.name CONTAINS diabetes\n",
    "  where = setConstraints(\n",
    "    paths = c(\"Gene.organism.name\", \"Gene.diseases.name\"),\n",
    "    operators = c(\"=\", \"CONTAINS\"),\n",
    "    values = list(\"Homo sapiens\",\"diabetes\")\n",
    "  )\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Okay, we've set the query up, so now let's actually run it:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene.primaryIdentifier</th><th scope=col>Gene.symbol</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>100188782</td><td>NIDDM4 </td></tr>\n",
       "\t<tr><td>1056     </td><td>CEL    </td></tr>\n",
       "\t<tr><td>10644    </td><td>IGF2BP2</td></tr>\n",
       "\t<tr><td>11132    </td><td>CAPN10 </td></tr>\n",
       "\t<tr><td>1234     </td><td>CCR5   </td></tr>\n",
       "\t<tr><td>1493     </td><td>CTLA4  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       " Gene.primaryIdentifier & Gene.symbol\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 100188782 & NIDDM4 \\\\\n",
       "\t 1056      & CEL    \\\\\n",
       "\t 10644     & IGF2BP2\\\\\n",
       "\t 11132     & CAPN10 \\\\\n",
       "\t 1234      & CCR5   \\\\\n",
       "\t 1493      & CTLA4  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| Gene.primaryIdentifier &lt;chr&gt; | Gene.symbol &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 100188782 | NIDDM4  |\n",
       "| 1056      | CEL     |\n",
       "| 10644     | IGF2BP2 |\n",
       "| 11132     | CAPN10  |\n",
       "| 1234      | CCR5    |\n",
       "| 1493      | CTLA4   |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene.primaryIdentifier Gene.symbol\n",
       "1 100188782              NIDDM4     \n",
       "2 1056                   CEL        \n",
       "3 10644                  IGF2BP2    \n",
       "4 11132                  CAPN10     \n",
       "5 1234                   CCR5       \n",
       "6 1493                   CTLA4      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "query1DiabetesResults <- runQuery(im,query1Diabetes)\n",
    "\n",
    "# and let's print out the first few results to make sure it looks like we'd expect:\n",
    "head(query1DiabetesResults)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Query 2: Pax6 targets that have high expression in the Pancreas\n",
    "\n",
    "This time we're creating another query, but with slightly more complex constraints. We're looking for genes that are in the public HumanMine list `PL_Pax6_Targets`, that are also expressed in the pancreas at a `High` or `Medium` level. \n",
    "\n",
    "We'll need a few more **constraints** than we did in Query 1:  \n",
    "1. all `Gene`s should be in the list `PL_Pax6_Targets`\n",
    "2. `Gene.proteinAtlasExpression.tissue.name` should be equal to `Pancreas`\n",
    "3. `Gene.proteinAtlasExpression.level` should be set to `High` OR `Medium`. This will require two constraints, one for each of medium and high.\n",
    "\n",
    "We'd also like to see a few more **columns** this time: \n",
    "1. The `Gene`'s `primaryIdentifier` and `symbol`\n",
    "2. The following expression data from Protein Atlas: \n",
    "    - `Gene.proteinAtlasExpression.cellType`\n",
    "    - `Gene.proteinAtlasExpression.level`\n",
    "    - `Gene.proteinAtlasExpression.tissue.name`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a new query\n",
    "query2UpInPancreas = newQuery(\n",
    "  #here we're choosing which columns of data we'd like to see\n",
    "  view = c(\"Gene.primaryIdentifier\",\n",
    "             \"Gene.symbol\",\n",
    "             \"Gene.proteinAtlasExpression.cellType\",\n",
    "             \"Gene.proteinAtlasExpression.level\",\n",
    "             \"Gene.proteinAtlasExpression.tissue.name\"\n",
    "  ),\n",
    "  # set the logic for constraints (see the function below for this to make sense)\n",
    "  constraintLogic = \"A and (B or C) and D\"\n",
    ")\n",
    "\n",
    "# If we ran the query above, it'd show us *all* genes and their expression. \n",
    "# Let's narrow it down a little by constraining it to genes that are of interest\n",
    "query2UpInPancreasConstraint = setConstraints(\n",
    "  paths = c(\"Gene\", \n",
    "            \"Gene.proteinAtlasExpression.level\", \n",
    "            \"Gene.proteinAtlasExpression.level\", \n",
    "            \"Gene.proteinAtlasExpression.tissue.name\"),\n",
    "  operators = c(\"IN\", rep(\"=\", 2), \"=\"),\n",
    "  # each constraint is automatically given a code, allowing us to manipulate the \n",
    "  # logic for the constraint. \n",
    "  # Below, the constraints are set to codes A, B, C, D in order, \n",
    "  #  e.g. Code A: \"Gene\" should be \"IN\" the list named \"PL_DiabetesGenes\"\n",
    "  #       Code B: \"Gene.proteinAtlasExpression.level\" should be equal to \"Medium\"\n",
    "  #       Code C: \"Gene.proteinAtlasExpression.level\" should be equal to \"High\"\n",
    "  #       Code D: \"Gene.proteinAtlasExpression.tissue.name\" should be equal to Pancreas\"\n",
    "  # \n",
    "  # Now, you might be thinking \"how can the expression level be equal to both Medium AND High?\"\n",
    "  # and the answer is - it can't, but take a quick look at the constraintLogic we set earlier - \n",
    "  # (B or C) makes it clear that we want one or the other (but not, for instance, Low) \n",
    "  values = list(\"PL_Pax6_Targets\", \"Medium\", \"High\", \"Pancreas\")\n",
    ")\n",
    "\n",
    "# Add the constraint to our expressed pancreas query (previously we just _defined_ the constraint)\n",
    "query2UpInPancreas$where <- query2UpInPancreasConstraint"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remember, that was just setting up the query - we haven't run it yet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene.primaryIdentifier</th><th scope=col>Gene.symbol</th><th scope=col>Gene.proteinAtlasExpression.cellType</th><th scope=col>Gene.proteinAtlasExpression.level</th><th scope=col>Gene.proteinAtlasExpression.tissue.name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>10097</td><td>ACTR2</td><td>exocrine glandular cells</td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>10097</td><td>ACTR2</td><td>islets of Langerhans    </td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>10196</td><td>PRMT3</td><td>exocrine glandular cells</td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>10196</td><td>PRMT3</td><td>islets of Langerhans    </td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>1121 </td><td>CHM  </td><td>exocrine glandular cells</td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>1121 </td><td>CHM  </td><td>islets of Langerhans    </td><td>Medium</td><td>Pancreas</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       " Gene.primaryIdentifier & Gene.symbol & Gene.proteinAtlasExpression.cellType & Gene.proteinAtlasExpression.level & Gene.proteinAtlasExpression.tissue.name\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 10097 & ACTR2 & exocrine glandular cells & Medium & Pancreas\\\\\n",
       "\t 10097 & ACTR2 & islets of Langerhans     & Medium & Pancreas\\\\\n",
       "\t 10196 & PRMT3 & exocrine glandular cells & Medium & Pancreas\\\\\n",
       "\t 10196 & PRMT3 & islets of Langerhans     & Medium & Pancreas\\\\\n",
       "\t 1121  & CHM   & exocrine glandular cells & Medium & Pancreas\\\\\n",
       "\t 1121  & CHM   & islets of Langerhans     & Medium & Pancreas\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| Gene.primaryIdentifier &lt;chr&gt; | Gene.symbol &lt;chr&gt; | Gene.proteinAtlasExpression.cellType &lt;chr&gt; | Gene.proteinAtlasExpression.level &lt;chr&gt; | Gene.proteinAtlasExpression.tissue.name &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 10097 | ACTR2 | exocrine glandular cells | Medium | Pancreas |\n",
       "| 10097 | ACTR2 | islets of Langerhans     | Medium | Pancreas |\n",
       "| 10196 | PRMT3 | exocrine glandular cells | Medium | Pancreas |\n",
       "| 10196 | PRMT3 | islets of Langerhans     | Medium | Pancreas |\n",
       "| 1121  | CHM   | exocrine glandular cells | Medium | Pancreas |\n",
       "| 1121  | CHM   | islets of Langerhans     | Medium | Pancreas |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene.primaryIdentifier Gene.symbol Gene.proteinAtlasExpression.cellType\n",
       "1 10097                  ACTR2       exocrine glandular cells            \n",
       "2 10097                  ACTR2       islets of Langerhans                \n",
       "3 10196                  PRMT3       exocrine glandular cells            \n",
       "4 10196                  PRMT3       islets of Langerhans                \n",
       "5 1121                   CHM         exocrine glandular cells            \n",
       "6 1121                   CHM         islets of Langerhans                \n",
       "  Gene.proteinAtlasExpression.level Gene.proteinAtlasExpression.tissue.name\n",
       "1 Medium                            Pancreas                               \n",
       "2 Medium                            Pancreas                               \n",
       "3 Medium                            Pancreas                               \n",
       "4 Medium                            Pancreas                               \n",
       "5 Medium                            Pancreas                               \n",
       "6 Medium                            Pancreas                               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Now we have the query set up the way we want, let's actually *run* the query! \n",
    "query2UpInPancreasResults <-  runQuery(im = im, qry = query2UpInPancreas)\n",
    "\n",
    "# Show me the first few results please! \n",
    "head(query2UpInPancreasResults) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intersection: Which genes overlap in Query1 and Query2?\n",
    "\n",
    "Let's check which genes are in BOTH lists that we've created. To do this, we'll strip down the columns we have to just primary identifiers, and then run a list intersect function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'100188782'</li>\n",
       "\t<li>'1056'</li>\n",
       "\t<li>'10644'</li>\n",
       "\t<li>'11132'</li>\n",
       "\t<li>'1234'</li>\n",
       "\t<li>'1493'</li>\n",
       "\t<li>'1636'</li>\n",
       "\t<li>'169026'</li>\n",
       "\t<li>'169792'</li>\n",
       "\t<li>'2056'</li>\n",
       "\t<li>'208'</li>\n",
       "\t<li>'26060'</li>\n",
       "\t<li>'26191'</li>\n",
       "\t<li>'2642'</li>\n",
       "\t<li>'2645'</li>\n",
       "\t<li>'2820'</li>\n",
       "\t<li>'3077'</li>\n",
       "\t<li>'3159'</li>\n",
       "\t<li>'3172'</li>\n",
       "\t<li>'346171'</li>\n",
       "\t<li>'3557'</li>\n",
       "\t<li>'3559'</li>\n",
       "\t<li>'3569'</li>\n",
       "\t<li>'359'</li>\n",
       "\t<li>'3630'</li>\n",
       "\t<li>'3643'</li>\n",
       "\t<li>'3651'</li>\n",
       "\t<li>'3667'</li>\n",
       "\t<li>'3710'</li>\n",
       "\t<li>'3767'</li>\n",
       "\t<li>'387082'</li>\n",
       "\t<li>'389692'</li>\n",
       "\t<li>'3990'</li>\n",
       "\t<li>'4544'</li>\n",
       "\t<li>'4760'</li>\n",
       "\t<li>'5078'</li>\n",
       "\t<li>'50943'</li>\n",
       "\t<li>'50982'</li>\n",
       "\t<li>'51124'</li>\n",
       "\t<li>'5167'</li>\n",
       "\t<li>'5325'</li>\n",
       "\t<li>'5444'</li>\n",
       "\t<li>'5468'</li>\n",
       "\t<li>'5506'</li>\n",
       "\t<li>'551'</li>\n",
       "\t<li>'554'</li>\n",
       "\t<li>'5611'</li>\n",
       "\t<li>'56729'</li>\n",
       "\t<li>'57061'</li>\n",
       "\t<li>'5770'</li>\n",
       "\t<li>'640'</li>\n",
       "\t<li>'6514'</li>\n",
       "\t<li>'6648'</li>\n",
       "\t<li>'6833'</li>\n",
       "\t<li>'6927'</li>\n",
       "\t<li>'6928'</li>\n",
       "\t<li>'6934'</li>\n",
       "\t<li>'7422'</li>\n",
       "\t<li>'7466'</li>\n",
       "\t<li>'8462'</li>\n",
       "\t<li>'8660'</li>\n",
       "\t<li>'9451'</li>\n",
       "\t<li>'9479'</li>\n",
       "\t<li>'9882'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '100188782'\n",
       "\\item '1056'\n",
       "\\item '10644'\n",
       "\\item '11132'\n",
       "\\item '1234'\n",
       "\\item '1493'\n",
       "\\item '1636'\n",
       "\\item '169026'\n",
       "\\item '169792'\n",
       "\\item '2056'\n",
       "\\item '208'\n",
       "\\item '26060'\n",
       "\\item '26191'\n",
       "\\item '2642'\n",
       "\\item '2645'\n",
       "\\item '2820'\n",
       "\\item '3077'\n",
       "\\item '3159'\n",
       "\\item '3172'\n",
       "\\item '346171'\n",
       "\\item '3557'\n",
       "\\item '3559'\n",
       "\\item '3569'\n",
       "\\item '359'\n",
       "\\item '3630'\n",
       "\\item '3643'\n",
       "\\item '3651'\n",
       "\\item '3667'\n",
       "\\item '3710'\n",
       "\\item '3767'\n",
       "\\item '387082'\n",
       "\\item '389692'\n",
       "\\item '3990'\n",
       "\\item '4544'\n",
       "\\item '4760'\n",
       "\\item '5078'\n",
       "\\item '50943'\n",
       "\\item '50982'\n",
       "\\item '51124'\n",
       "\\item '5167'\n",
       "\\item '5325'\n",
       "\\item '5444'\n",
       "\\item '5468'\n",
       "\\item '5506'\n",
       "\\item '551'\n",
       "\\item '554'\n",
       "\\item '5611'\n",
       "\\item '56729'\n",
       "\\item '57061'\n",
       "\\item '5770'\n",
       "\\item '640'\n",
       "\\item '6514'\n",
       "\\item '6648'\n",
       "\\item '6833'\n",
       "\\item '6927'\n",
       "\\item '6928'\n",
       "\\item '6934'\n",
       "\\item '7422'\n",
       "\\item '7466'\n",
       "\\item '8462'\n",
       "\\item '8660'\n",
       "\\item '9451'\n",
       "\\item '9479'\n",
       "\\item '9882'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '100188782'\n",
       "2. '1056'\n",
       "3. '10644'\n",
       "4. '11132'\n",
       "5. '1234'\n",
       "6. '1493'\n",
       "7. '1636'\n",
       "8. '169026'\n",
       "9. '169792'\n",
       "10. '2056'\n",
       "11. '208'\n",
       "12. '26060'\n",
       "13. '26191'\n",
       "14. '2642'\n",
       "15. '2645'\n",
       "16. '2820'\n",
       "17. '3077'\n",
       "18. '3159'\n",
       "19. '3172'\n",
       "20. '346171'\n",
       "21. '3557'\n",
       "22. '3559'\n",
       "23. '3569'\n",
       "24. '359'\n",
       "25. '3630'\n",
       "26. '3643'\n",
       "27. '3651'\n",
       "28. '3667'\n",
       "29. '3710'\n",
       "30. '3767'\n",
       "31. '387082'\n",
       "32. '389692'\n",
       "33. '3990'\n",
       "34. '4544'\n",
       "35. '4760'\n",
       "36. '5078'\n",
       "37. '50943'\n",
       "38. '50982'\n",
       "39. '51124'\n",
       "40. '5167'\n",
       "41. '5325'\n",
       "42. '5444'\n",
       "43. '5468'\n",
       "44. '5506'\n",
       "45. '551'\n",
       "46. '554'\n",
       "47. '5611'\n",
       "48. '56729'\n",
       "49. '57061'\n",
       "50. '5770'\n",
       "51. '640'\n",
       "52. '6514'\n",
       "53. '6648'\n",
       "54. '6833'\n",
       "55. '6927'\n",
       "56. '6928'\n",
       "57. '6934'\n",
       "58. '7422'\n",
       "59. '7466'\n",
       "60. '8462'\n",
       "61. '8660'\n",
       "62. '9451'\n",
       "63. '9479'\n",
       "64. '9882'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"100188782\" \"1056\"      \"10644\"     \"11132\"     \"1234\"      \"1493\"     \n",
       " [7] \"1636\"      \"169026\"    \"169792\"    \"2056\"      \"208\"       \"26060\"    \n",
       "[13] \"26191\"     \"2642\"      \"2645\"      \"2820\"      \"3077\"      \"3159\"     \n",
       "[19] \"3172\"      \"346171\"    \"3557\"      \"3559\"      \"3569\"      \"359\"      \n",
       "[25] \"3630\"      \"3643\"      \"3651\"      \"3667\"      \"3710\"      \"3767\"     \n",
       "[31] \"387082\"    \"389692\"    \"3990\"      \"4544\"      \"4760\"      \"5078\"     \n",
       "[37] \"50943\"     \"50982\"     \"51124\"     \"5167\"      \"5325\"      \"5444\"     \n",
       "[43] \"5468\"      \"5506\"      \"551\"       \"554\"       \"5611\"      \"56729\"    \n",
       "[49] \"57061\"     \"5770\"      \"640\"       \"6514\"      \"6648\"      \"6833\"     \n",
       "[55] \"6927\"      \"6928\"      \"6934\"      \"7422\"      \"7466\"      \"8462\"     \n",
       "[61] \"8660\"      \"9451\"      \"9479\"      \"9882\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"---\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'10097'</li>\n",
       "\t<li>'10097'</li>\n",
       "\t<li>'10196'</li>\n",
       "\t<li>'10196'</li>\n",
       "\t<li>'1121'</li>\n",
       "\t<li>'1121'</li>\n",
       "\t<li>'11340'</li>\n",
       "\t<li>'114818'</li>\n",
       "\t<li>'115286'</li>\n",
       "\t<li>'116039'</li>\n",
       "\t<li>'124540'</li>\n",
       "\t<li>'124540'</li>\n",
       "\t<li>'140609'</li>\n",
       "\t<li>'140609'</li>\n",
       "\t<li>'143098'</li>\n",
       "\t<li>'143098'</li>\n",
       "\t<li>'143187'</li>\n",
       "\t<li>'144100'</li>\n",
       "\t<li>'144100'</li>\n",
       "\t<li>'145282'</li>\n",
       "\t<li>'145282'</li>\n",
       "\t<li>'1478'</li>\n",
       "\t<li>'1478'</li>\n",
       "\t<li>'171425'</li>\n",
       "\t<li>'171425'</li>\n",
       "\t<li>'1848'</li>\n",
       "\t<li>'201255'</li>\n",
       "\t<li>'2045'</li>\n",
       "\t<li>'2045'</li>\n",
       "\t<li>'2048'</li>\n",
       "\t<li>'219402'</li>\n",
       "\t<li>'219402'</li>\n",
       "\t<li>'22827'</li>\n",
       "\t<li>'22827'</li>\n",
       "\t<li>'22920'</li>\n",
       "\t<li>'22920'</li>\n",
       "\t<li>'23085'</li>\n",
       "\t<li>'23133'</li>\n",
       "\t<li>'23133'</li>\n",
       "\t<li>'23200'</li>\n",
       "\t<li>'23328'</li>\n",
       "\t<li>'23328'</li>\n",
       "\t<li>'23414'</li>\n",
       "\t<li>'23414'</li>\n",
       "\t<li>'23613'</li>\n",
       "\t<li>'2494'</li>\n",
       "\t<li>'2494'</li>\n",
       "\t<li>'25925'</li>\n",
       "\t<li>'2619'</li>\n",
       "\t<li>'26468'</li>\n",
       "\t<li>'26468'</li>\n",
       "\t<li>'26610'</li>\n",
       "\t<li>'26610'</li>\n",
       "\t<li>'27072'</li>\n",
       "\t<li>'27316'</li>\n",
       "\t<li>'27316'</li>\n",
       "\t<li>'2736'</li>\n",
       "\t<li>'2736'</li>\n",
       "\t<li>'28964'</li>\n",
       "\t<li>'2971'</li>\n",
       "\t<li>'29880'</li>\n",
       "\t<li>'3172'</li>\n",
       "\t<li>'3187'</li>\n",
       "\t<li>'3187'</li>\n",
       "\t<li>'3205'</li>\n",
       "\t<li>'3217'</li>\n",
       "\t<li>'3233'</li>\n",
       "\t<li>'3233'</li>\n",
       "\t<li>'3397'</li>\n",
       "\t<li>'3397'</li>\n",
       "\t<li>'340371'</li>\n",
       "\t<li>'340371'</li>\n",
       "\t<li>'3491'</li>\n",
       "\t<li>'359948'</li>\n",
       "\t<li>'359948'</li>\n",
       "\t<li>'3615'</li>\n",
       "\t<li>'3615'</li>\n",
       "\t<li>'3728'</li>\n",
       "\t<li>'3728'</li>\n",
       "\t<li>'374868'</li>\n",
       "\t<li>'3842'</li>\n",
       "\t<li>'3842'</li>\n",
       "\t<li>'389072'</li>\n",
       "\t<li>'389072'</li>\n",
       "\t<li>'3983'</li>\n",
       "\t<li>'4082'</li>\n",
       "\t<li>'4082'</li>\n",
       "\t<li>'4211'</li>\n",
       "\t<li>'4212'</li>\n",
       "\t<li>'4212'</li>\n",
       "\t<li>'4301'</li>\n",
       "\t<li>'4756'</li>\n",
       "\t<li>'476'</li>\n",
       "\t<li>'4774'</li>\n",
       "\t<li>'490'</li>\n",
       "\t<li>'490'</li>\n",
       "\t<li>'5033'</li>\n",
       "\t<li>'5080'</li>\n",
       "\t<li>'5087'</li>\n",
       "\t<li>'5087'</li>\n",
       "\t<li>'5090'</li>\n",
       "\t<li>'5090'</li>\n",
       "\t<li>'51319'</li>\n",
       "\t<li>'51319'</li>\n",
       "\t<li>'5144'</li>\n",
       "\t<li>'5150'</li>\n",
       "\t<li>'51741'</li>\n",
       "\t<li>'51741'</li>\n",
       "\t<li>'51776'</li>\n",
       "\t<li>'5195'</li>\n",
       "\t<li>'5195'</li>\n",
       "\t<li>'5339'</li>\n",
       "\t<li>'5422'</li>\n",
       "\t<li>'54674'</li>\n",
       "\t<li>'54681'</li>\n",
       "\t<li>'54681'</li>\n",
       "\t<li>'54828'</li>\n",
       "\t<li>'54828'</li>\n",
       "\t<li>'55084'</li>\n",
       "\t<li>'55084'</li>\n",
       "\t<li>'55120'</li>\n",
       "\t<li>'55120'</li>\n",
       "\t<li>'55137'</li>\n",
       "\t<li>'55137'</li>\n",
       "\t<li>'55152'</li>\n",
       "\t<li>'55152'</li>\n",
       "\t<li>'5566'</li>\n",
       "\t<li>'5566'</li>\n",
       "\t<li>'55703'</li>\n",
       "\t<li>'5607'</li>\n",
       "\t<li>'57117'</li>\n",
       "\t<li>'57117'</li>\n",
       "\t<li>'5727'</li>\n",
       "\t<li>'5727'</li>\n",
       "\t<li>'57565'</li>\n",
       "\t<li>'57565'</li>\n",
       "\t<li>'57616'</li>\n",
       "\t<li>'5862'</li>\n",
       "\t<li>'5862'</li>\n",
       "\t<li>'6210'</li>\n",
       "\t<li>'63877'</li>\n",
       "\t<li>'6428'</li>\n",
       "\t<li>'6428'</li>\n",
       "\t<li>'65125'</li>\n",
       "\t<li>'657'</li>\n",
       "\t<li>'6809'</li>\n",
       "\t<li>'6809'</li>\n",
       "\t<li>'6840'</li>\n",
       "\t<li>'6902'</li>\n",
       "\t<li>'6902'</li>\n",
       "\t<li>'6907'</li>\n",
       "\t<li>'6907'</li>\n",
       "\t<li>'6928'</li>\n",
       "\t<li>'6934'</li>\n",
       "\t<li>'6934'</li>\n",
       "\t<li>'6938'</li>\n",
       "\t<li>'7110'</li>\n",
       "\t<li>'7110'</li>\n",
       "\t<li>'7444'</li>\n",
       "\t<li>'7533'</li>\n",
       "\t<li>'79767'</li>\n",
       "\t<li>'79971'</li>\n",
       "\t<li>'80700'</li>\n",
       "\t<li>'8339'</li>\n",
       "\t<li>'8339'</li>\n",
       "\t<li>'83938'</li>\n",
       "\t<li>'83989'</li>\n",
       "\t<li>'83989'</li>\n",
       "\t<li>'84193'</li>\n",
       "\t<li>'84529'</li>\n",
       "\t<li>'84618'</li>\n",
       "\t<li>'84669'</li>\n",
       "\t<li>'84669'</li>\n",
       "\t<li>'84858'</li>\n",
       "\t<li>'84858'</li>\n",
       "\t<li>'8543'</li>\n",
       "\t<li>'8543'</li>\n",
       "\t<li>'8667'</li>\n",
       "\t<li>'8667'</li>\n",
       "\t<li>'8891'</li>\n",
       "\t<li>'8891'</li>\n",
       "\t<li>'9039'</li>\n",
       "\t<li>'9039'</li>\n",
       "\t<li>'905'</li>\n",
       "\t<li>'905'</li>\n",
       "\t<li>'93986'</li>\n",
       "\t<li>'93986'</li>\n",
       "\t<li>'9678'</li>\n",
       "\t<li>'9686'</li>\n",
       "\t<li>'9690'</li>\n",
       "\t<li>'9690'</li>\n",
       "\t<li>'9792'</li>\n",
       "\t<li>'987'</li>\n",
       "\t<li>'9922'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '10097'\n",
       "\\item '10097'\n",
       "\\item '10196'\n",
       "\\item '10196'\n",
       "\\item '1121'\n",
       "\\item '1121'\n",
       "\\item '11340'\n",
       "\\item '114818'\n",
       "\\item '115286'\n",
       "\\item '116039'\n",
       "\\item '124540'\n",
       "\\item '124540'\n",
       "\\item '140609'\n",
       "\\item '140609'\n",
       "\\item '143098'\n",
       "\\item '143098'\n",
       "\\item '143187'\n",
       "\\item '144100'\n",
       "\\item '144100'\n",
       "\\item '145282'\n",
       "\\item '145282'\n",
       "\\item '1478'\n",
       "\\item '1478'\n",
       "\\item '171425'\n",
       "\\item '171425'\n",
       "\\item '1848'\n",
       "\\item '201255'\n",
       "\\item '2045'\n",
       "\\item '2045'\n",
       "\\item '2048'\n",
       "\\item '219402'\n",
       "\\item '219402'\n",
       "\\item '22827'\n",
       "\\item '22827'\n",
       "\\item '22920'\n",
       "\\item '22920'\n",
       "\\item '23085'\n",
       "\\item '23133'\n",
       "\\item '23133'\n",
       "\\item '23200'\n",
       "\\item '23328'\n",
       "\\item '23328'\n",
       "\\item '23414'\n",
       "\\item '23414'\n",
       "\\item '23613'\n",
       "\\item '2494'\n",
       "\\item '2494'\n",
       "\\item '25925'\n",
       "\\item '2619'\n",
       "\\item '26468'\n",
       "\\item '26468'\n",
       "\\item '26610'\n",
       "\\item '26610'\n",
       "\\item '27072'\n",
       "\\item '27316'\n",
       "\\item '27316'\n",
       "\\item '2736'\n",
       "\\item '2736'\n",
       "\\item '28964'\n",
       "\\item '2971'\n",
       "\\item '29880'\n",
       "\\item '3172'\n",
       "\\item '3187'\n",
       "\\item '3187'\n",
       "\\item '3205'\n",
       "\\item '3217'\n",
       "\\item '3233'\n",
       "\\item '3233'\n",
       "\\item '3397'\n",
       "\\item '3397'\n",
       "\\item '340371'\n",
       "\\item '340371'\n",
       "\\item '3491'\n",
       "\\item '359948'\n",
       "\\item '359948'\n",
       "\\item '3615'\n",
       "\\item '3615'\n",
       "\\item '3728'\n",
       "\\item '3728'\n",
       "\\item '374868'\n",
       "\\item '3842'\n",
       "\\item '3842'\n",
       "\\item '389072'\n",
       "\\item '389072'\n",
       "\\item '3983'\n",
       "\\item '4082'\n",
       "\\item '4082'\n",
       "\\item '4211'\n",
       "\\item '4212'\n",
       "\\item '4212'\n",
       "\\item '4301'\n",
       "\\item '4756'\n",
       "\\item '476'\n",
       "\\item '4774'\n",
       "\\item '490'\n",
       "\\item '490'\n",
       "\\item '5033'\n",
       "\\item '5080'\n",
       "\\item '5087'\n",
       "\\item '5087'\n",
       "\\item '5090'\n",
       "\\item '5090'\n",
       "\\item '51319'\n",
       "\\item '51319'\n",
       "\\item '5144'\n",
       "\\item '5150'\n",
       "\\item '51741'\n",
       "\\item '51741'\n",
       "\\item '51776'\n",
       "\\item '5195'\n",
       "\\item '5195'\n",
       "\\item '5339'\n",
       "\\item '5422'\n",
       "\\item '54674'\n",
       "\\item '54681'\n",
       "\\item '54681'\n",
       "\\item '54828'\n",
       "\\item '54828'\n",
       "\\item '55084'\n",
       "\\item '55084'\n",
       "\\item '55120'\n",
       "\\item '55120'\n",
       "\\item '55137'\n",
       "\\item '55137'\n",
       "\\item '55152'\n",
       "\\item '55152'\n",
       "\\item '5566'\n",
       "\\item '5566'\n",
       "\\item '55703'\n",
       "\\item '5607'\n",
       "\\item '57117'\n",
       "\\item '57117'\n",
       "\\item '5727'\n",
       "\\item '5727'\n",
       "\\item '57565'\n",
       "\\item '57565'\n",
       "\\item '57616'\n",
       "\\item '5862'\n",
       "\\item '5862'\n",
       "\\item '6210'\n",
       "\\item '63877'\n",
       "\\item '6428'\n",
       "\\item '6428'\n",
       "\\item '65125'\n",
       "\\item '657'\n",
       "\\item '6809'\n",
       "\\item '6809'\n",
       "\\item '6840'\n",
       "\\item '6902'\n",
       "\\item '6902'\n",
       "\\item '6907'\n",
       "\\item '6907'\n",
       "\\item '6928'\n",
       "\\item '6934'\n",
       "\\item '6934'\n",
       "\\item '6938'\n",
       "\\item '7110'\n",
       "\\item '7110'\n",
       "\\item '7444'\n",
       "\\item '7533'\n",
       "\\item '79767'\n",
       "\\item '79971'\n",
       "\\item '80700'\n",
       "\\item '8339'\n",
       "\\item '8339'\n",
       "\\item '83938'\n",
       "\\item '83989'\n",
       "\\item '83989'\n",
       "\\item '84193'\n",
       "\\item '84529'\n",
       "\\item '84618'\n",
       "\\item '84669'\n",
       "\\item '84669'\n",
       "\\item '84858'\n",
       "\\item '84858'\n",
       "\\item '8543'\n",
       "\\item '8543'\n",
       "\\item '8667'\n",
       "\\item '8667'\n",
       "\\item '8891'\n",
       "\\item '8891'\n",
       "\\item '9039'\n",
       "\\item '9039'\n",
       "\\item '905'\n",
       "\\item '905'\n",
       "\\item '93986'\n",
       "\\item '93986'\n",
       "\\item '9678'\n",
       "\\item '9686'\n",
       "\\item '9690'\n",
       "\\item '9690'\n",
       "\\item '9792'\n",
       "\\item '987'\n",
       "\\item '9922'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '10097'\n",
       "2. '10097'\n",
       "3. '10196'\n",
       "4. '10196'\n",
       "5. '1121'\n",
       "6. '1121'\n",
       "7. '11340'\n",
       "8. '114818'\n",
       "9. '115286'\n",
       "10. '116039'\n",
       "11. '124540'\n",
       "12. '124540'\n",
       "13. '140609'\n",
       "14. '140609'\n",
       "15. '143098'\n",
       "16. '143098'\n",
       "17. '143187'\n",
       "18. '144100'\n",
       "19. '144100'\n",
       "20. '145282'\n",
       "21. '145282'\n",
       "22. '1478'\n",
       "23. '1478'\n",
       "24. '171425'\n",
       "25. '171425'\n",
       "26. '1848'\n",
       "27. '201255'\n",
       "28. '2045'\n",
       "29. '2045'\n",
       "30. '2048'\n",
       "31. '219402'\n",
       "32. '219402'\n",
       "33. '22827'\n",
       "34. '22827'\n",
       "35. '22920'\n",
       "36. '22920'\n",
       "37. '23085'\n",
       "38. '23133'\n",
       "39. '23133'\n",
       "40. '23200'\n",
       "41. '23328'\n",
       "42. '23328'\n",
       "43. '23414'\n",
       "44. '23414'\n",
       "45. '23613'\n",
       "46. '2494'\n",
       "47. '2494'\n",
       "48. '25925'\n",
       "49. '2619'\n",
       "50. '26468'\n",
       "51. '26468'\n",
       "52. '26610'\n",
       "53. '26610'\n",
       "54. '27072'\n",
       "55. '27316'\n",
       "56. '27316'\n",
       "57. '2736'\n",
       "58. '2736'\n",
       "59. '28964'\n",
       "60. '2971'\n",
       "61. '29880'\n",
       "62. '3172'\n",
       "63. '3187'\n",
       "64. '3187'\n",
       "65. '3205'\n",
       "66. '3217'\n",
       "67. '3233'\n",
       "68. '3233'\n",
       "69. '3397'\n",
       "70. '3397'\n",
       "71. '340371'\n",
       "72. '340371'\n",
       "73. '3491'\n",
       "74. '359948'\n",
       "75. '359948'\n",
       "76. '3615'\n",
       "77. '3615'\n",
       "78. '3728'\n",
       "79. '3728'\n",
       "80. '374868'\n",
       "81. '3842'\n",
       "82. '3842'\n",
       "83. '389072'\n",
       "84. '389072'\n",
       "85. '3983'\n",
       "86. '4082'\n",
       "87. '4082'\n",
       "88. '4211'\n",
       "89. '4212'\n",
       "90. '4212'\n",
       "91. '4301'\n",
       "92. '4756'\n",
       "93. '476'\n",
       "94. '4774'\n",
       "95. '490'\n",
       "96. '490'\n",
       "97. '5033'\n",
       "98. '5080'\n",
       "99. '5087'\n",
       "100. '5087'\n",
       "101. '5090'\n",
       "102. '5090'\n",
       "103. '51319'\n",
       "104. '51319'\n",
       "105. '5144'\n",
       "106. '5150'\n",
       "107. '51741'\n",
       "108. '51741'\n",
       "109. '51776'\n",
       "110. '5195'\n",
       "111. '5195'\n",
       "112. '5339'\n",
       "113. '5422'\n",
       "114. '54674'\n",
       "115. '54681'\n",
       "116. '54681'\n",
       "117. '54828'\n",
       "118. '54828'\n",
       "119. '55084'\n",
       "120. '55084'\n",
       "121. '55120'\n",
       "122. '55120'\n",
       "123. '55137'\n",
       "124. '55137'\n",
       "125. '55152'\n",
       "126. '55152'\n",
       "127. '5566'\n",
       "128. '5566'\n",
       "129. '55703'\n",
       "130. '5607'\n",
       "131. '57117'\n",
       "132. '57117'\n",
       "133. '5727'\n",
       "134. '5727'\n",
       "135. '57565'\n",
       "136. '57565'\n",
       "137. '57616'\n",
       "138. '5862'\n",
       "139. '5862'\n",
       "140. '6210'\n",
       "141. '63877'\n",
       "142. '6428'\n",
       "143. '6428'\n",
       "144. '65125'\n",
       "145. '657'\n",
       "146. '6809'\n",
       "147. '6809'\n",
       "148. '6840'\n",
       "149. '6902'\n",
       "150. '6902'\n",
       "151. '6907'\n",
       "152. '6907'\n",
       "153. '6928'\n",
       "154. '6934'\n",
       "155. '6934'\n",
       "156. '6938'\n",
       "157. '7110'\n",
       "158. '7110'\n",
       "159. '7444'\n",
       "160. '7533'\n",
       "161. '79767'\n",
       "162. '79971'\n",
       "163. '80700'\n",
       "164. '8339'\n",
       "165. '8339'\n",
       "166. '83938'\n",
       "167. '83989'\n",
       "168. '83989'\n",
       "169. '84193'\n",
       "170. '84529'\n",
       "171. '84618'\n",
       "172. '84669'\n",
       "173. '84669'\n",
       "174. '84858'\n",
       "175. '84858'\n",
       "176. '8543'\n",
       "177. '8543'\n",
       "178. '8667'\n",
       "179. '8667'\n",
       "180. '8891'\n",
       "181. '8891'\n",
       "182. '9039'\n",
       "183. '9039'\n",
       "184. '905'\n",
       "185. '905'\n",
       "186. '93986'\n",
       "187. '93986'\n",
       "188. '9678'\n",
       "189. '9686'\n",
       "190. '9690'\n",
       "191. '9690'\n",
       "192. '9792'\n",
       "193. '987'\n",
       "194. '9922'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"10097\"  \"10097\"  \"10196\"  \"10196\"  \"1121\"   \"1121\"   \"11340\"  \"114818\"\n",
       "  [9] \"115286\" \"116039\" \"124540\" \"124540\" \"140609\" \"140609\" \"143098\" \"143098\"\n",
       " [17] \"143187\" \"144100\" \"144100\" \"145282\" \"145282\" \"1478\"   \"1478\"   \"171425\"\n",
       " [25] \"171425\" \"1848\"   \"201255\" \"2045\"   \"2045\"   \"2048\"   \"219402\" \"219402\"\n",
       " [33] \"22827\"  \"22827\"  \"22920\"  \"22920\"  \"23085\"  \"23133\"  \"23133\"  \"23200\" \n",
       " [41] \"23328\"  \"23328\"  \"23414\"  \"23414\"  \"23613\"  \"2494\"   \"2494\"   \"25925\" \n",
       " [49] \"2619\"   \"26468\"  \"26468\"  \"26610\"  \"26610\"  \"27072\"  \"27316\"  \"27316\" \n",
       " [57] \"2736\"   \"2736\"   \"28964\"  \"2971\"   \"29880\"  \"3172\"   \"3187\"   \"3187\"  \n",
       " [65] \"3205\"   \"3217\"   \"3233\"   \"3233\"   \"3397\"   \"3397\"   \"340371\" \"340371\"\n",
       " [73] \"3491\"   \"359948\" \"359948\" \"3615\"   \"3615\"   \"3728\"   \"3728\"   \"374868\"\n",
       " [81] \"3842\"   \"3842\"   \"389072\" \"389072\" \"3983\"   \"4082\"   \"4082\"   \"4211\"  \n",
       " [89] \"4212\"   \"4212\"   \"4301\"   \"4756\"   \"476\"    \"4774\"   \"490\"    \"490\"   \n",
       " [97] \"5033\"   \"5080\"   \"5087\"   \"5087\"   \"5090\"   \"5090\"   \"51319\"  \"51319\" \n",
       "[105] \"5144\"   \"5150\"   \"51741\"  \"51741\"  \"51776\"  \"5195\"   \"5195\"   \"5339\"  \n",
       "[113] \"5422\"   \"54674\"  \"54681\"  \"54681\"  \"54828\"  \"54828\"  \"55084\"  \"55084\" \n",
       "[121] \"55120\"  \"55120\"  \"55137\"  \"55137\"  \"55152\"  \"55152\"  \"5566\"   \"5566\"  \n",
       "[129] \"55703\"  \"5607\"   \"57117\"  \"57117\"  \"5727\"   \"5727\"   \"57565\"  \"57565\" \n",
       "[137] \"57616\"  \"5862\"   \"5862\"   \"6210\"   \"63877\"  \"6428\"   \"6428\"   \"65125\" \n",
       "[145] \"657\"    \"6809\"   \"6809\"   \"6840\"   \"6902\"   \"6902\"   \"6907\"   \"6907\"  \n",
       "[153] \"6928\"   \"6934\"   \"6934\"   \"6938\"   \"7110\"   \"7110\"   \"7444\"   \"7533\"  \n",
       "[161] \"79767\"  \"79971\"  \"80700\"  \"8339\"   \"8339\"   \"83938\"  \"83989\"  \"83989\" \n",
       "[169] \"84193\"  \"84529\"  \"84618\"  \"84669\"  \"84669\"  \"84858\"  \"84858\"  \"8543\"  \n",
       "[177] \"8543\"   \"8667\"   \"8667\"   \"8891\"   \"8891\"   \"9039\"   \"9039\"   \"905\"   \n",
       "[185] \"905\"    \"93986\"  \"93986\"  \"9678\"   \"9686\"   \"9690\"   \"9690\"   \"9792\"  \n",
       "[193] \"987\"    \"9922\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'3172'</li>\n",
       "\t<li>'6928'</li>\n",
       "\t<li>'6934'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '3172'\n",
       "\\item '6928'\n",
       "\\item '6934'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '3172'\n",
       "2. '6928'\n",
       "3. '6934'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"3172\" \"6928\" \"6934\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "primaryIdentifiers.diabetes <- query1DiabetesResults[[\"Gene.primaryIdentifier\"]]\n",
    "primaryIdentifiers.pancreas <- query2UpInPancreasResults[[\"Gene.primaryIdentifier\"]]\n",
    "\n",
    "primaryIdentifiers.diabetes\n",
    "print (\"---\")\n",
    "primaryIdentifiers.pancreas \n",
    "\n",
    "intersect(primaryIdentifiers.diabetes,primaryIdentifiers.pancreas)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
