{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# InterMineR Workshop Use Case\n",
    "\n",
    "\n",
    "\n",
    "We are going to re-create the workflow we did using the web interface using the R API.\n",
    "\n",
    "The basic steps are:  \n",
    "\n",
    "1. Load the InterMine library and choose an InterMine to query.\n",
    "2. Query 1: Diabetes Genes: Fetch a list of genes that are associated with diabetes\n",
    "3. Query 2: PAX6 + Pancreas: Fetch a list of genes that have medium or high expression in the pancreas and are in our PAX6 targets list\n",
    "4. Intersection: Find which genes are present in _both_ Query 1 and Query2.\n",
    "5. GWAS: Compare the intersection of the previous query with results from GWAS studies."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting started - Load InterMineR and choose an InterMine"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the InterMine library. If it's not already installed, visit\n",
    "https://bioconductor.org/packages/release/bioc/html/InterMineR.html and follow the instructions to install.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(InterMineR)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We want to query human data - so let's look and see what InterMines are available: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>BMAP</dt>\n",
       "\t\t<dd>'https://bmap.jgi.doe.gov/bmapmine/'</dd>\n",
       "\t<dt>BeanMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/beanmine'</dd>\n",
       "\t<dt>BovineMine</dt>\n",
       "\t\t<dd>'http://genomes.missouri.edu/bovinemine'</dd>\n",
       "\t<dt>CHOmine</dt>\n",
       "\t\t<dd>'https://chomine.boku.ac.at/chomine'</dd>\n",
       "\t<dt>ChickpeaMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/chickpeamine'</dd>\n",
       "\t<dt>CowpeaMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/cowpeamine'</dd>\n",
       "\t<dt>FlyMine</dt>\n",
       "\t\t<dd>'http://www.flymine.org/flymine'</dd>\n",
       "\t<dt>GrapeMine</dt>\n",
       "\t\t<dd>'http://urgi.versailles.inra.fr/GrapeMine'</dd>\n",
       "\t<dt>HumanMine</dt>\n",
       "\t\t<dd>'http://www.humanmine.org/humanmine'</dd>\n",
       "\t<dt>HymenopteraMine</dt>\n",
       "\t\t<dd>'http://hymenopteragenome.org/hymenopteramine'</dd>\n",
       "\t<dt>IndigoMine</dt>\n",
       "\t\t<dd>'http://www.cbrc.kaust.edu.sa/indigo'</dd>\n",
       "\t<dt>LegumeMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/legumemine'</dd>\n",
       "\t<dt>MaizeMine</dt>\n",
       "\t\t<dd>'http://maizemine.rnet.missouri.edu:8080/maizemine'</dd>\n",
       "\t<dt>MedicMine</dt>\n",
       "\t\t<dd>'http://medicmine.jcvi.org/medicmine'</dd>\n",
       "\t<dt>MitoMiner</dt>\n",
       "\t\t<dd>'http://mitominer.mrc-mbu.cam.ac.uk/release-4.0'</dd>\n",
       "\t<dt>ModMine</dt>\n",
       "\t\t<dd>'http://intermine.modencode.org/release-33'</dd>\n",
       "\t<dt>MouseMine</dt>\n",
       "\t\t<dd>'http://www.mousemine.org/mousemine'</dd>\n",
       "\t<dt>OakMine</dt>\n",
       "\t\t<dd>'https://urgi.versailles.inra.fr/OakMine_PM1N'</dd>\n",
       "\t<dt>PeanutMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/peanutmine'</dd>\n",
       "\t<dt>PhytoMine</dt>\n",
       "\t\t<dd>'https://phytozome.jgi.doe.gov/phytomine/'</dd>\n",
       "\t<dt>PlanMine</dt>\n",
       "\t\t<dd>'http://planmine.mpi-cbg.de/planmine'</dd>\n",
       "\t<dt>RatMine</dt>\n",
       "\t\t<dd>'http://ratmine.mcw.edu/ratmine'</dd>\n",
       "\t<dt>RepetDB</dt>\n",
       "\t\t<dd>'http://urgi.versailles.inra.fr/repetdb'</dd>\n",
       "\t<dt>SoyMine</dt>\n",
       "\t\t<dd>'https://mines.legumeinfo.org/soymine'</dd>\n",
       "\t<dt>TargetMine</dt>\n",
       "\t\t<dd>'https://targetmine.mizuguchilab.org/targetmine'</dd>\n",
       "\t<dt>ThaleMine</dt>\n",
       "\t\t<dd>'https://apps.araport.org/thalemine'</dd>\n",
       "\t<dt>WheatMine</dt>\n",
       "\t\t<dd>'https://urgi.versailles.inra.fr/WheatMine'</dd>\n",
       "\t<dt>WormMine</dt>\n",
       "\t\t<dd>'http://intermine.wormbase.org/tools/wormmine/'</dd>\n",
       "\t<dt>XenMine</dt>\n",
       "\t\t<dd>'http://www.xenmine.org/xenmine'</dd>\n",
       "\t<dt>YeastMine</dt>\n",
       "\t\t<dd>'https://yeastmine.yeastgenome.org/yeastmine'</dd>\n",
       "\t<dt>ZebrafishMine</dt>\n",
       "\t\t<dd>'http://zebrafishmine.org'</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[BMAP] 'https://bmap.jgi.doe.gov/bmapmine/'\n",
       "\\item[BeanMine] 'https://mines.legumeinfo.org/beanmine'\n",
       "\\item[BovineMine] 'http://genomes.missouri.edu/bovinemine'\n",
       "\\item[CHOmine] 'https://chomine.boku.ac.at/chomine'\n",
       "\\item[ChickpeaMine] 'https://mines.legumeinfo.org/chickpeamine'\n",
       "\\item[CowpeaMine] 'https://mines.legumeinfo.org/cowpeamine'\n",
       "\\item[FlyMine] 'http://www.flymine.org/flymine'\n",
       "\\item[GrapeMine] 'http://urgi.versailles.inra.fr/GrapeMine'\n",
       "\\item[HumanMine] 'http://www.humanmine.org/humanmine'\n",
       "\\item[HymenopteraMine] 'http://hymenopteragenome.org/hymenopteramine'\n",
       "\\item[IndigoMine] 'http://www.cbrc.kaust.edu.sa/indigo'\n",
       "\\item[LegumeMine] 'https://mines.legumeinfo.org/legumemine'\n",
       "\\item[MaizeMine] 'http://maizemine.rnet.missouri.edu:8080/maizemine'\n",
       "\\item[MedicMine] 'http://medicmine.jcvi.org/medicmine'\n",
       "\\item[MitoMiner] 'http://mitominer.mrc-mbu.cam.ac.uk/release-4.0'\n",
       "\\item[ModMine] 'http://intermine.modencode.org/release-33'\n",
       "\\item[MouseMine] 'http://www.mousemine.org/mousemine'\n",
       "\\item[OakMine] 'https://urgi.versailles.inra.fr/OakMine\\_PM1N'\n",
       "\\item[PeanutMine] 'https://mines.legumeinfo.org/peanutmine'\n",
       "\\item[PhytoMine] 'https://phytozome.jgi.doe.gov/phytomine/'\n",
       "\\item[PlanMine] 'http://planmine.mpi-cbg.de/planmine'\n",
       "\\item[RatMine] 'http://ratmine.mcw.edu/ratmine'\n",
       "\\item[RepetDB] 'http://urgi.versailles.inra.fr/repetdb'\n",
       "\\item[SoyMine] 'https://mines.legumeinfo.org/soymine'\n",
       "\\item[TargetMine] 'https://targetmine.mizuguchilab.org/targetmine'\n",
       "\\item[ThaleMine] 'https://apps.araport.org/thalemine'\n",
       "\\item[WheatMine] 'https://urgi.versailles.inra.fr/WheatMine'\n",
       "\\item[WormMine] 'http://intermine.wormbase.org/tools/wormmine/'\n",
       "\\item[XenMine] 'http://www.xenmine.org/xenmine'\n",
       "\\item[YeastMine] 'https://yeastmine.yeastgenome.org/yeastmine'\n",
       "\\item[ZebrafishMine] 'http://zebrafishmine.org'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "BMAP\n",
       ":   'https://bmap.jgi.doe.gov/bmapmine/'BeanMine\n",
       ":   'https://mines.legumeinfo.org/beanmine'BovineMine\n",
       ":   'http://genomes.missouri.edu/bovinemine'CHOmine\n",
       ":   'https://chomine.boku.ac.at/chomine'ChickpeaMine\n",
       ":   'https://mines.legumeinfo.org/chickpeamine'CowpeaMine\n",
       ":   'https://mines.legumeinfo.org/cowpeamine'FlyMine\n",
       ":   'http://www.flymine.org/flymine'GrapeMine\n",
       ":   'http://urgi.versailles.inra.fr/GrapeMine'HumanMine\n",
       ":   'http://www.humanmine.org/humanmine'HymenopteraMine\n",
       ":   'http://hymenopteragenome.org/hymenopteramine'IndigoMine\n",
       ":   'http://www.cbrc.kaust.edu.sa/indigo'LegumeMine\n",
       ":   'https://mines.legumeinfo.org/legumemine'MaizeMine\n",
       ":   'http://maizemine.rnet.missouri.edu:8080/maizemine'MedicMine\n",
       ":   'http://medicmine.jcvi.org/medicmine'MitoMiner\n",
       ":   'http://mitominer.mrc-mbu.cam.ac.uk/release-4.0'ModMine\n",
       ":   'http://intermine.modencode.org/release-33'MouseMine\n",
       ":   'http://www.mousemine.org/mousemine'OakMine\n",
       ":   'https://urgi.versailles.inra.fr/OakMine_PM1N'PeanutMine\n",
       ":   'https://mines.legumeinfo.org/peanutmine'PhytoMine\n",
       ":   'https://phytozome.jgi.doe.gov/phytomine/'PlanMine\n",
       ":   'http://planmine.mpi-cbg.de/planmine'RatMine\n",
       ":   'http://ratmine.mcw.edu/ratmine'RepetDB\n",
       ":   'http://urgi.versailles.inra.fr/repetdb'SoyMine\n",
       ":   'https://mines.legumeinfo.org/soymine'TargetMine\n",
       ":   'https://targetmine.mizuguchilab.org/targetmine'ThaleMine\n",
       ":   'https://apps.araport.org/thalemine'WheatMine\n",
       ":   'https://urgi.versailles.inra.fr/WheatMine'WormMine\n",
       ":   'http://intermine.wormbase.org/tools/wormmine/'XenMine\n",
       ":   'http://www.xenmine.org/xenmine'YeastMine\n",
       ":   'https://yeastmine.yeastgenome.org/yeastmine'ZebrafishMine\n",
       ":   'http://zebrafishmine.org'\n",
       "\n"
      ],
      "text/plain": [
       "                                               BMAP \n",
       "               \"https://bmap.jgi.doe.gov/bmapmine/\" \n",
       "                                           BeanMine \n",
       "            \"https://mines.legumeinfo.org/beanmine\" \n",
       "                                         BovineMine \n",
       "           \"http://genomes.missouri.edu/bovinemine\" \n",
       "                                            CHOmine \n",
       "               \"https://chomine.boku.ac.at/chomine\" \n",
       "                                       ChickpeaMine \n",
       "        \"https://mines.legumeinfo.org/chickpeamine\" \n",
       "                                         CowpeaMine \n",
       "          \"https://mines.legumeinfo.org/cowpeamine\" \n",
       "                                            FlyMine \n",
       "                   \"http://www.flymine.org/flymine\" \n",
       "                                          GrapeMine \n",
       "         \"http://urgi.versailles.inra.fr/GrapeMine\" \n",
       "                                          HumanMine \n",
       "               \"http://www.humanmine.org/humanmine\" \n",
       "                                    HymenopteraMine \n",
       "     \"http://hymenopteragenome.org/hymenopteramine\" \n",
       "                                         IndigoMine \n",
       "              \"http://www.cbrc.kaust.edu.sa/indigo\" \n",
       "                                         LegumeMine \n",
       "          \"https://mines.legumeinfo.org/legumemine\" \n",
       "                                          MaizeMine \n",
       "\"http://maizemine.rnet.missouri.edu:8080/maizemine\" \n",
       "                                          MedicMine \n",
       "              \"http://medicmine.jcvi.org/medicmine\" \n",
       "                                          MitoMiner \n",
       "   \"http://mitominer.mrc-mbu.cam.ac.uk/release-4.0\" \n",
       "                                            ModMine \n",
       "        \"http://intermine.modencode.org/release-33\" \n",
       "                                          MouseMine \n",
       "               \"http://www.mousemine.org/mousemine\" \n",
       "                                            OakMine \n",
       "     \"https://urgi.versailles.inra.fr/OakMine_PM1N\" \n",
       "                                         PeanutMine \n",
       "          \"https://mines.legumeinfo.org/peanutmine\" \n",
       "                                          PhytoMine \n",
       "         \"https://phytozome.jgi.doe.gov/phytomine/\" \n",
       "                                           PlanMine \n",
       "              \"http://planmine.mpi-cbg.de/planmine\" \n",
       "                                            RatMine \n",
       "                   \"http://ratmine.mcw.edu/ratmine\" \n",
       "                                            RepetDB \n",
       "           \"http://urgi.versailles.inra.fr/repetdb\" \n",
       "                                            SoyMine \n",
       "             \"https://mines.legumeinfo.org/soymine\" \n",
       "                                         TargetMine \n",
       "   \"https://targetmine.mizuguchilab.org/targetmine\" \n",
       "                                          ThaleMine \n",
       "               \"https://apps.araport.org/thalemine\" \n",
       "                                          WheatMine \n",
       "        \"https://urgi.versailles.inra.fr/WheatMine\" \n",
       "                                           WormMine \n",
       "    \"http://intermine.wormbase.org/tools/wormmine/\" \n",
       "                                            XenMine \n",
       "                   \"http://www.xenmine.org/xenmine\" \n",
       "                                          YeastMine \n",
       "      \"https://yeastmine.yeastgenome.org/yeastmine\" \n",
       "                                      ZebrafishMine \n",
       "                         \"http://zebrafishmine.org\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "listMines()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Okay, let's select HumanMine from the list:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>HumanMine:</strong> 'http://www.humanmine.org/humanmine'"
      ],
      "text/latex": [
       "\\textbf{HumanMine:} 'http://www.humanmine.org/humanmine'"
      ],
      "text/markdown": [
       "**HumanMine:** 'http://www.humanmine.org/humanmine'"
      ],
      "text/plain": [
       "                           HumanMine \n",
       "\"http://www.humanmine.org/humanmine\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "humanMine <- listMines()[\"HumanMine\"] #select humanmine\n",
    "humanMine #print out the value to see what's inside"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Okay, now let's tell InterMineR that we want to use HumanMine for our queries.\n",
    "\n",
    "**Important:** you'll need an API token for this part so you can access your HumanMine account. You can get your token by logging into [HumanMine](http://www.humanmine.org/) and going to the account details tab within MyMine. Cut and paste your token into the code below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "im <- initInterMine(mine=humanMine, \"YOUR TOKEN HERE\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### First Query: Diabetes Genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Our first query will be to select all human genes that are associate with diabetes. This will require two constraints:\n",
    "\n",
    "1. Ensure all genes returned are `Home sapiens` genes (HumanMine contains some non-human genes for homology query purposes)\n",
    "2. Restrict results to genes that are associated with `diabetes`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "query1Diabetes <- setQuery( \n",
    "  # here we're choosing which columns of data we'd like to see\n",
    "  select = c(\"Gene.primaryIdentifier\", \"Gene.symbol\"),\n",
    "    # set the logic for constraints. The first constraint is the first path+operator+value, \n",
    "    # e.g. Gene.organism.name = Homo sapiens, and the second constraint is the combination \n",
    "    # of the second path+operator+value, e.g. Gene.diseases.name CONTAINS diabetes\n",
    "  where = setConstraints(\n",
    "    paths = c(\"Gene.organism.name\", \"Gene.diseases.name\"),\n",
    "    operators = c(\"=\", \"CONTAINS\"),\n",
    "    values = list(\"Homo sapiens\",\"diabetes\")\n",
    "  )\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Question to ponder:** why did we use `=` for our Homo sapiens constraint, but `CONTAINS` for our diabetes constraint?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Anyway, we've set the query up, so now let's actually run it:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene.primaryIdentifier</th><th scope=col>Gene.symbol</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>100188782</td><td>NIDDM4 </td></tr>\n",
       "\t<tr><td>1056     </td><td>CEL    </td></tr>\n",
       "\t<tr><td>10644    </td><td>IGF2BP2</td></tr>\n",
       "\t<tr><td>11132    </td><td>CAPN10 </td></tr>\n",
       "\t<tr><td>1234     </td><td>CCR5   </td></tr>\n",
       "\t<tr><td>1493     </td><td>CTLA4  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       " Gene.primaryIdentifier & Gene.symbol\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 100188782 & NIDDM4 \\\\\n",
       "\t 1056      & CEL    \\\\\n",
       "\t 10644     & IGF2BP2\\\\\n",
       "\t 11132     & CAPN10 \\\\\n",
       "\t 1234      & CCR5   \\\\\n",
       "\t 1493      & CTLA4  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| Gene.primaryIdentifier &lt;chr&gt; | Gene.symbol &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 100188782 | NIDDM4  |\n",
       "| 1056      | CEL     |\n",
       "| 10644     | IGF2BP2 |\n",
       "| 11132     | CAPN10  |\n",
       "| 1234      | CCR5    |\n",
       "| 1493      | CTLA4   |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene.primaryIdentifier Gene.symbol\n",
       "1 100188782              NIDDM4     \n",
       "2 1056                   CEL        \n",
       "3 10644                  IGF2BP2    \n",
       "4 11132                  CAPN10     \n",
       "5 1234                   CCR5       \n",
       "6 1493                   CTLA4      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "query1DiabetesResults <- runQuery(im,query1Diabetes)\n",
    "\n",
    "# and let's print out the first few results to make sure it looks like we'd expect:\n",
    "head(query1DiabetesResults)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Query 2: Pax6 targets that have high expression in the Pancreas\n",
    "\n",
    "This time we're creating another query, but with slightly more complex constraints. We're looking for genes that are in the public HumanMine list `PL_Pax6_Targets`, that are also expressed in the pancreas at a `High` or `Medium` level. \n",
    "\n",
    "We'll need a few more **constraints** than we did in Query 1:  \n",
    "1. all `Gene`s should be in the list `PL_Pax6_Targets`\n",
    "2. `Gene.proteinAtlasExpression.tissue.name` should be equal to `Pancreas`\n",
    "3. `Gene.proteinAtlasExpression.level` should be set to `High` OR `Medium`. This will require two constraints, one for each of medium and high.\n",
    "\n",
    "We'd also like to see a few more **columns** this time: \n",
    "1. The `Gene`'s `primaryIdentifier` and `symbol`\n",
    "2. The following expression data from Protein Atlas: \n",
    "    - `Gene.proteinAtlasExpression.cellType`\n",
    "    - `Gene.proteinAtlasExpression.level`\n",
    "    - `Gene.proteinAtlasExpression.tissue.name`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We don't want to see *all* genes and their expression. \n",
    "# Let's narrow it down a little by constraining it to genes that are of interest\n",
    "query2UpInPancreasConstraint = setConstraints(\n",
    "  paths = c(\"Gene\", \n",
    "            \"Gene.proteinAtlasExpression.level\", \n",
    "            \"Gene.proteinAtlasExpression.level\", \n",
    "            \"Gene.proteinAtlasExpression.tissue.name\"),\n",
    "  operators = c(\"IN\", rep(\"=\", 3)),\n",
    "  # each constraint is automatically given a code, allowing us to manipulate the \n",
    "  # logic for the constraint. \n",
    "  #  So for us, constraints are set to codes A, B, C, D in order, \n",
    "  #  e.g. Code A: \"Gene\" should be \"IN\" the list named \"PL_DiabetesGenes\"\n",
    "  #       Code B: \"Gene.proteinAtlasExpression.level\" should be equal to \"Medium\"\n",
    "  #       Code C: \"Gene.proteinAtlasExpression.level\" should be equal to \"High\"\n",
    "  #       Code D: \"Gene.proteinAtlasExpression.tissue.name\" should be equal to Pancreas\"\n",
    "  # \n",
    "  # Now, you might be thinking \"how can the expression level be equal to both Medium\n",
    "  # AND High?\" The answer is - it can't, but take a quick look at the constraintLogic\n",
    "  # we will set in the next code cell for an explanation\n",
    "  values = list(\"PL_Pax6_Targets\", \"Medium\", \"High\", \"Pancreas\")\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Excellent - we've defined the constraints we want. We still need to choose which columns to view."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a new query\n",
    "query2UpInPancreas = newQuery(\n",
    "  # Choose which columns of data we'd like to see\n",
    "  view = c(\"Gene.primaryIdentifier\",\n",
    "             \"Gene.symbol\",\n",
    "             \"Gene.proteinAtlasExpression.cellType\",\n",
    "             \"Gene.proteinAtlasExpression.level\",\n",
    "             \"Gene.proteinAtlasExpression.tissue.name\"\n",
    "  ),\n",
    "  # set the logic for constraints. This means our pancreas expression level \n",
    "  # is EITHER Medium (B) or High (C), but not both.\n",
    "  # --\n",
    "  # Note: Constraint logic only needs to be set if you wish to use OR. All other\n",
    "  # constraints have AND logic applied by default. \n",
    "  constraintLogic = \"A and (B or C) and D\"\n",
    ")\n",
    "\n",
    "# Add the constraint to our expressed pancreas query (previously we just _defined_ the constraint)\n",
    "query2UpInPancreas$where <- query2UpInPancreasConstraint"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remember, that was just setting up the query - we haven't run it yet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 6 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene.primaryIdentifier</th><th scope=col>Gene.symbol</th><th scope=col>Gene.proteinAtlasExpression.cellType</th><th scope=col>Gene.proteinAtlasExpression.level</th><th scope=col>Gene.proteinAtlasExpression.tissue.name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>10097</td><td>ACTR2</td><td>exocrine glandular cells</td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>10097</td><td>ACTR2</td><td>islets of Langerhans    </td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>10196</td><td>PRMT3</td><td>exocrine glandular cells</td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>10196</td><td>PRMT3</td><td>islets of Langerhans    </td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>1121 </td><td>CHM  </td><td>exocrine glandular cells</td><td>Medium</td><td>Pancreas</td></tr>\n",
       "\t<tr><td>1121 </td><td>CHM  </td><td>islets of Langerhans    </td><td>Medium</td><td>Pancreas</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       " Gene.primaryIdentifier & Gene.symbol & Gene.proteinAtlasExpression.cellType & Gene.proteinAtlasExpression.level & Gene.proteinAtlasExpression.tissue.name\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 10097 & ACTR2 & exocrine glandular cells & Medium & Pancreas\\\\\n",
       "\t 10097 & ACTR2 & islets of Langerhans     & Medium & Pancreas\\\\\n",
       "\t 10196 & PRMT3 & exocrine glandular cells & Medium & Pancreas\\\\\n",
       "\t 10196 & PRMT3 & islets of Langerhans     & Medium & Pancreas\\\\\n",
       "\t 1121  & CHM   & exocrine glandular cells & Medium & Pancreas\\\\\n",
       "\t 1121  & CHM   & islets of Langerhans     & Medium & Pancreas\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 5\n",
       "\n",
       "| Gene.primaryIdentifier &lt;chr&gt; | Gene.symbol &lt;chr&gt; | Gene.proteinAtlasExpression.cellType &lt;chr&gt; | Gene.proteinAtlasExpression.level &lt;chr&gt; | Gene.proteinAtlasExpression.tissue.name &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 10097 | ACTR2 | exocrine glandular cells | Medium | Pancreas |\n",
       "| 10097 | ACTR2 | islets of Langerhans     | Medium | Pancreas |\n",
       "| 10196 | PRMT3 | exocrine glandular cells | Medium | Pancreas |\n",
       "| 10196 | PRMT3 | islets of Langerhans     | Medium | Pancreas |\n",
       "| 1121  | CHM   | exocrine glandular cells | Medium | Pancreas |\n",
       "| 1121  | CHM   | islets of Langerhans     | Medium | Pancreas |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene.primaryIdentifier Gene.symbol Gene.proteinAtlasExpression.cellType\n",
       "1 10097                  ACTR2       exocrine glandular cells            \n",
       "2 10097                  ACTR2       islets of Langerhans                \n",
       "3 10196                  PRMT3       exocrine glandular cells            \n",
       "4 10196                  PRMT3       islets of Langerhans                \n",
       "5 1121                   CHM         exocrine glandular cells            \n",
       "6 1121                   CHM         islets of Langerhans                \n",
       "  Gene.proteinAtlasExpression.level Gene.proteinAtlasExpression.tissue.name\n",
       "1 Medium                            Pancreas                               \n",
       "2 Medium                            Pancreas                               \n",
       "3 Medium                            Pancreas                               \n",
       "4 Medium                            Pancreas                               \n",
       "5 Medium                            Pancreas                               \n",
       "6 Medium                            Pancreas                               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Now we have the query set up the way we want, let's actually *run* the query! \n",
    "query2UpInPancreasResults <-  runQuery(im = im, qry = query2UpInPancreas)\n",
    "\n",
    "# Show me the first few results please! \n",
    "head(query2UpInPancreasResults) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intersection: Which genes overlap in Query1 and Query2?\n",
    "\n",
    "Let's check which genes are in BOTH lists that we've created. To do this, we'll strip down the columns we have to just primary identifiers, and then run a list intersect function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"3172\" \"6928\" \"6934\"\n"
     ]
    }
   ],
   "source": [
    "# Extract the primaryIdentifier columns from query1 (diabetes genes) and query 2 (upexpressed in pancreas)\n",
    "primaryIdentifiers.diabetes <- query1DiabetesResults[[\"Gene.primaryIdentifier\"]]\n",
    "primaryIdentifiers.pancreas <- query2UpInPancreasResults[[\"Gene.primaryIdentifier\"]]\n",
    "\n",
    "# Find the intersection of the two lists of primary identifiers\n",
    "diabetesAndPancreasGenes <- intersect(primaryIdentifiers.diabetes,primaryIdentifiers.pancreas)\n",
    "\n",
    "# Show the results\n",
    "print(diabetesAndPancreasGenes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GWAS: Compare the intersection above with results from GWAS studies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finally, we fed the intersected list from above back into another query to see if there was any association of these genes with diabetes phenotypes according to GWAS studies. Note that we now start our query from the `GWAS` class:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, we set up the constraints. The last three constraints are the \n",
    "# diabetesAndPancreas result genes from our last query.\n",
    "query3GWASConstraints <- setConstraints(\n",
    "    paths = c(\"GWAS.results.pValue\", \n",
    "              \"GWAS.results.phenotype\",\n",
    "              # using rep so we don't have to type this three times... \n",
    "              rep(\"GWAS.results.associatedGenes.primaryIdentifier\",3)\n",
    "             ),\n",
    "    operators = c(\"<=\", \n",
    "                  \"CONTAINS\",\n",
    "                  rep(\"=\",3)),\n",
    "    values = list(\"1e-04\",   #A\n",
    "                  \"diabetes\",#B \n",
    "                  \"3172\",    #C\n",
    "                  \"6928\",    #D\n",
    "                  \"6934\")    #E\n",
    "  )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we've set our constraints up nicely, let's choose which columns we want to view. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "query3GWAS <- newQuery( \n",
    "  # Quite a few columns this time!\n",
    "  view = c(\"GWAS.results.associatedGenes.primaryIdentifier\",\n",
    "    \"GWAS.results.associatedGenes.symbol\", \"GWAS.results.associatedGenes.name\",\n",
    "    \"GWAS.results.SNP.primaryIdentifier\", \"GWAS.results.pValue\", \"GWAS.results.phenotype\",\n",
    "    \"GWAS.firstAuthor\", \"GWAS.name\", \"GWAS.publication.pubMedId\",\n",
    "    \"GWAS.results.associatedGenes.organism.shortName\"),\n",
    "    # set the logic for constraints. Remember that we want our results\n",
    "    # to include any one of the three genes we found in the list of diabetes+pancreas genes\n",
    "    # so we need to use some OR logic.\n",
    "  constraintLogic = \"A and B and (C or D or E)\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Add the constraints to the query, and then run it..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add constraint\n",
    "query3GWAS$where <- query3GWASConstraints\n",
    "#run query\n",
    "query3GWASResults <- runQuery(im, query3GWAS)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, let's view those results..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 46 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GWAS.results.associatedGenes.primaryIdentifier</th><th scope=col>GWAS.results.associatedGenes.symbol</th><th scope=col>GWAS.results.associatedGenes.name</th><th scope=col>GWAS.results.SNP.primaryIdentifier</th><th scope=col>GWAS.results.pValue</th><th scope=col>GWAS.results.phenotype</th><th scope=col>GWAS.firstAuthor</th><th scope=col>GWAS.name</th><th scope=col>GWAS.publication.pubMedId</th><th scope=col>GWAS.results.associatedGenes.organism.shortName</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>3172</td><td>HNF4A </td><td>hepatocyte nuclear factor 4 alpha</td><td>rs4812829  </td><td>3.0E-10 </td><td>Type 2 diabetes                 </td><td>Kooner JS                             </td><td>Genome-wide association study in individuals of South Asian ancestry identifies six new type 2 diabetes susceptibility loci.                        </td><td>21874001</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>3172</td><td>HNF4A </td><td>hepatocyte nuclear factor 4 alpha</td><td>rs4812829  </td><td>4.0E-10 </td><td>Type 2 diabetes                 </td><td>Zhao W                                </td><td>Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                          </td><td>28869590</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>3172</td><td>HNF4A </td><td>hepatocyte nuclear factor 4 alpha</td><td>rs4812829  </td><td>5.0E-8  </td><td>Type 2 diabetes                 </td><td>Mahajan A                             </td><td>Genome-wide trans-ancestry meta-analysis provides insight into the genetic architecture of type 2 diabetes susceptibility.                          </td><td>24509480</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs117229942</td><td>4.0E-11 </td><td>Type 2 diabetes                 </td><td>Xue A                                 </td><td>Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                 </td><td>30054458</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs34872471 </td><td>1.0E-94 </td><td>Type 2 diabetes                 </td><td>Bonas-Guarch S                        </td><td>Re-analysis of public genetic data reveals a rare X-chromosomal variant associated with type 2 diabetes.                                            </td><td>29358691</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs34872471 </td><td>6.0E-53 </td><td>Type 2 diabetes                 </td><td>Cook JP                               </td><td>Multi-ethnic genome-wide association study identifies novel locus for type 2 diabetes susceptibility.                                               </td><td>27189021</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs34872471 </td><td>3.0E-23 </td><td>Type 2 diabetes                 </td><td>Imamura M                             </td><td>Genome-wide association studies in the Japanese population identify seven novel loci for type 2 diabetes.                                           </td><td>26818947</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs34872471 </td><td>8.0E-8  </td><td>Type 2 diabetes                 </td><td>Ghassibe-Sabbagh M                    </td><td>T2DM GWAS in the Lebanese population confirms the role of TCF7L2 and CDKAL1 in disease susceptibility.                                              </td><td>25483131</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs4506565  </td><td>5.0E-12 </td><td>Type 2 diabetes                 </td><td>Wellcome Trust Case Control Consortium</td><td>Genome-wide association study of 14,000 cases of seven common diseases and 3,000 shared controls.                                                   </td><td>17554300</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs4918796  </td><td>4.0E-13 </td><td>Type 2 diabetes                 </td><td>Xue A                                 </td><td>Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                 </td><td>30054458</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7901695  </td><td>1.0E-48 </td><td>Type 2 diabetes                 </td><td>Zeggini E                             </td><td>Replication of genome-wide association signals in UK samples reveals risk loci for type 2 diabetes.                                                 </td><td>17463249</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7901695  </td><td>1.0E-6  </td><td>Type 2 diabetes                 </td><td>Lettre G                              </td><td>Genome-wide association study of coronary heart disease and its risk factors in 8,090 African Americans: the NHLBI CARe Project.                    </td><td>21347282</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>0.0     </td><td>Type 2 diabetes                 </td><td>Xue A                                 </td><td>Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                 </td><td>30054458</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>1.0E-219</td><td>Type 2 diabetes                 </td><td>Zhao W                                </td><td>Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                          </td><td>28869590</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>2.0E-184</td><td>Type 2 diabetes                 </td><td>Zhao W                                </td><td>Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                          </td><td>28869590</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>1.0E-139</td><td>Type 2 diabetes                 </td><td>Morris AP                             </td><td>Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                            </td><td>22885922</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>4.0E-94 </td><td>Type 2 diabetes                 </td><td>Ng MC                                 </td><td>Meta-analysis of genome-wide association studies in African Americans provides insights into the genetic architecture of type 2 diabetes.           </td><td>25102180</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>6.0E-92 </td><td>Type 2 diabetes                 </td><td>Morris AP                             </td><td>Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                            </td><td>22885922</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>8.0E-75 </td><td>Type 2 diabetes                 </td><td>Mahajan A                             </td><td>Genome-wide trans-ancestry meta-analysis provides insight into the genetic architecture of type 2 diabetes susceptibility.                          </td><td>24509480</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>9.0E-75 </td><td>Type 2 diabetes                 </td><td>Saxena R                              </td><td>Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        </td><td>23300278</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>4.0E-73 </td><td>Type 2 diabetes                 </td><td>Morris AP                             </td><td>Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                            </td><td>22885922</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>2.0E-51 </td><td>Type 2 diabetes                 </td><td>Voight BF                             </td><td>Twelve type 2 diabetes susceptibility loci identified through large-scale association analysis.                                                     </td><td>20581827</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>1.0E-48 </td><td>Type 2 diabetes                 </td><td>Saxena R                              </td><td>Genome-wide association analysis identifies loci for type 2 diabetes and triglyceride levels.                                                       </td><td>17463246</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>7.0E-45 </td><td>Type 2 diabetes                 </td><td>Wood AR                               </td><td>Variants in the FTO and CDKAL1 loci have recessive effects on risk of obesity and type 2 diabetes, respectively.                                    </td><td>26961502</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>5.0E-44 </td><td>Type 2 diabetes                 </td><td>Ng MC                                 </td><td>Meta-analysis of genome-wide association studies in African Americans provides insights into the genetic architecture of type 2 diabetes.           </td><td>25102180</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>2.0E-40 </td><td>Type 2 diabetes                 </td><td>Perry JR                              </td><td>Stratifying type 2 diabetes cases by BMI identifies genetic risk variants in LAMA1 and enrichment for risk variants in lean compared to obese cases.</td><td>22693455</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>2.0E-38 </td><td>Type 2 diabetes                 </td><td>Saxena R                              </td><td>Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        </td><td>23300278</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>1.0E-35 </td><td>Type 2 diabetes                 </td><td>Tabassum R                            </td><td>Genome-wide association study for type 2 diabetes in Indians identifies a new susceptibility locus at 2q21.                                         </td><td>23209189</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>3.0E-35 </td><td>Type 2 diabetes                 </td><td>Saxena R                              </td><td>Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        </td><td>23300278</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>2.0E-34 </td><td>Type 2 diabetes                 </td><td>Sladek R                              </td><td>A genome-wide association study identifies novel risk loci for type 2 diabetes.                                                                     </td><td>17293876</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>1.0E-30 </td><td>Type 2 diabetes and other traits</td><td>Rung J                                </td><td>Genetic variant near IRS1 is associated with type 2 diabetes, insulin resistance and hyperinsulinemia.                                              </td><td>19734900</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>9.0E-30 </td><td>Type 2 diabetes                 </td><td>Timpson NJ                            </td><td>Adiposity-related heterogeneity in patterns of type 2 diabetes susceptibility observed in genome-wide association data.                             </td><td>19056611</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>3.0E-23 </td><td>Type 2 diabetes                 </td><td>Zeggini E                             </td><td>Meta-analysis of genome-wide association data and large-scale replication identifies additional susceptibility loci for type 2 diabetes.            </td><td>18372903</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>3.0E-23 </td><td>Type 2 diabetes                 </td><td>Zhao W                                </td><td>Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                          </td><td>28869590</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>6.0E-22 </td><td>Type 2 diabetes                 </td><td>Saxena R                              </td><td>Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        </td><td>23300278</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>4.0E-21 </td><td>Type 2 diabetes                 </td><td>Perry JR                              </td><td>Stratifying type 2 diabetes cases by BMI identifies genetic risk variants in LAMA1 and enrichment for risk variants in lean compared to obese cases.</td><td>22693455</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>3.0E-19 </td><td>Type 2 diabetes                 </td><td>Saxena R                              </td><td>Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        </td><td>23300278</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>6.0E-16 </td><td>Type 2 diabetes                 </td><td>Timpson NJ                            </td><td>Adiposity-related heterogeneity in patterns of type 2 diabetes susceptibility observed in genome-wide association data.                             </td><td>19056611</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>2.0E-15 </td><td>Type 2 diabetes                 </td><td>Hara K                                </td><td>Genome-wide association study identifies three novel loci for type 2 diabetes.                                                                      </td><td>23945395</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>2.0E-15 </td><td>Type 2 diabetes                 </td><td>Kho AN                                </td><td>Use of diverse electronic medical record systems to identify genetic risk for type 2 diabetes within a genome-wide association study.               </td><td>22101970</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>1.0E-14 </td><td>Type 2 diabetes                 </td><td>Williams AL                           </td><td>Sequence variants in SLC16A11 are a common risk factor for type 2 diabetes in Mexico.                                                               </td><td>24390345</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>8.0E-12 </td><td>Type 2 diabetes                 </td><td>Takeuchi F                            </td><td>Confirmation of multiple risk Loci and genetic impacts by a genome-wide association study of type 2 diabetes in the Japanese population.            </td><td>19401414</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>1.0E-11 </td><td>Type 2 diabetes                 </td><td>Qi Q                                  </td><td>Genetics of Type 2 Diabetes in U.S. Hispanic/Latino Individuals: Results From the Hispanic Community Health Study/Study of Latinos (HCHS/SOL).      </td><td>28254843</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>2.0E-10 </td><td>Type 2 diabetes                 </td><td>Steinthorsdottir V                    </td><td>A variant in CDKAL1 influences insulin response and risk of type 2 diabetes.                                                                        </td><td>17460697</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>1.0E-8  </td><td>Type 2 diabetes                 </td><td>Scott LJ                              </td><td>A genome-wide association study of type 2 diabetes in Finns detects multiple susceptibility variants.                                               </td><td>17463248</td><td>H. sapiens</td></tr>\n",
       "\t<tr><td>6934</td><td>TCF7L2</td><td>transcription factor 7 like 2    </td><td>rs7903146  </td><td>5.0E-8  </td><td>Type 2 diabetes                 </td><td>Salonen JT                            </td><td>Type 2 diabetes whole-genome association study in four populations: the DiaGen consortium.                                                          </td><td>17668382</td><td>H. sapiens</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 46 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       " GWAS.results.associatedGenes.primaryIdentifier & GWAS.results.associatedGenes.symbol & GWAS.results.associatedGenes.name & GWAS.results.SNP.primaryIdentifier & GWAS.results.pValue & GWAS.results.phenotype & GWAS.firstAuthor & GWAS.name & GWAS.publication.pubMedId & GWAS.results.associatedGenes.organism.shortName\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 3172 & HNF4A  & hepatocyte nuclear factor 4 alpha & rs4812829   & 3.0E-10  & Type 2 diabetes                  & Kooner JS                              & Genome-wide association study in individuals of South Asian ancestry identifies six new type 2 diabetes susceptibility loci.                         & 21874001 & H. sapiens\\\\\n",
       "\t 3172 & HNF4A  & hepatocyte nuclear factor 4 alpha & rs4812829   & 4.0E-10  & Type 2 diabetes                  & Zhao W                                 & Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                           & 28869590 & H. sapiens\\\\\n",
       "\t 3172 & HNF4A  & hepatocyte nuclear factor 4 alpha & rs4812829   & 5.0E-8   & Type 2 diabetes                  & Mahajan A                              & Genome-wide trans-ancestry meta-analysis provides insight into the genetic architecture of type 2 diabetes susceptibility.                           & 24509480 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs117229942 & 4.0E-11  & Type 2 diabetes                  & Xue A                                  & Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                  & 30054458 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs34872471  & 1.0E-94  & Type 2 diabetes                  & Bonas-Guarch S                         & Re-analysis of public genetic data reveals a rare X-chromosomal variant associated with type 2 diabetes.                                             & 29358691 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs34872471  & 6.0E-53  & Type 2 diabetes                  & Cook JP                                & Multi-ethnic genome-wide association study identifies novel locus for type 2 diabetes susceptibility.                                                & 27189021 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs34872471  & 3.0E-23  & Type 2 diabetes                  & Imamura M                              & Genome-wide association studies in the Japanese population identify seven novel loci for type 2 diabetes.                                            & 26818947 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs34872471  & 8.0E-8   & Type 2 diabetes                  & Ghassibe-Sabbagh M                     & T2DM GWAS in the Lebanese population confirms the role of TCF7L2 and CDKAL1 in disease susceptibility.                                               & 25483131 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs4506565   & 5.0E-12  & Type 2 diabetes                  & Wellcome Trust Case Control Consortium & Genome-wide association study of 14,000 cases of seven common diseases and 3,000 shared controls.                                                    & 17554300 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs4918796   & 4.0E-13  & Type 2 diabetes                  & Xue A                                  & Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                  & 30054458 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7901695   & 1.0E-48  & Type 2 diabetes                  & Zeggini E                              & Replication of genome-wide association signals in UK samples reveals risk loci for type 2 diabetes.                                                  & 17463249 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7901695   & 1.0E-6   & Type 2 diabetes                  & Lettre G                               & Genome-wide association study of coronary heart disease and its risk factors in 8,090 African Americans: the NHLBI CARe Project.                     & 21347282 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 0.0      & Type 2 diabetes                  & Xue A                                  & Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                  & 30054458 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 1.0E-219 & Type 2 diabetes                  & Zhao W                                 & Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                           & 28869590 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 2.0E-184 & Type 2 diabetes                  & Zhao W                                 & Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                           & 28869590 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 1.0E-139 & Type 2 diabetes                  & Morris AP                              & Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                             & 22885922 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 4.0E-94  & Type 2 diabetes                  & Ng MC                                  & Meta-analysis of genome-wide association studies in African Americans provides insights into the genetic architecture of type 2 diabetes.            & 25102180 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 6.0E-92  & Type 2 diabetes                  & Morris AP                              & Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                             & 22885922 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 8.0E-75  & Type 2 diabetes                  & Mahajan A                              & Genome-wide trans-ancestry meta-analysis provides insight into the genetic architecture of type 2 diabetes susceptibility.                           & 24509480 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 9.0E-75  & Type 2 diabetes                  & Saxena R                               & Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         & 23300278 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 4.0E-73  & Type 2 diabetes                  & Morris AP                              & Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                             & 22885922 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 2.0E-51  & Type 2 diabetes                  & Voight BF                              & Twelve type 2 diabetes susceptibility loci identified through large-scale association analysis.                                                      & 20581827 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 1.0E-48  & Type 2 diabetes                  & Saxena R                               & Genome-wide association analysis identifies loci for type 2 diabetes and triglyceride levels.                                                        & 17463246 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 7.0E-45  & Type 2 diabetes                  & Wood AR                                & Variants in the FTO and CDKAL1 loci have recessive effects on risk of obesity and type 2 diabetes, respectively.                                     & 26961502 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 5.0E-44  & Type 2 diabetes                  & Ng MC                                  & Meta-analysis of genome-wide association studies in African Americans provides insights into the genetic architecture of type 2 diabetes.            & 25102180 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 2.0E-40  & Type 2 diabetes                  & Perry JR                               & Stratifying type 2 diabetes cases by BMI identifies genetic risk variants in LAMA1 and enrichment for risk variants in lean compared to obese cases. & 22693455 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 2.0E-38  & Type 2 diabetes                  & Saxena R                               & Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         & 23300278 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 1.0E-35  & Type 2 diabetes                  & Tabassum R                             & Genome-wide association study for type 2 diabetes in Indians identifies a new susceptibility locus at 2q21.                                          & 23209189 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 3.0E-35  & Type 2 diabetes                  & Saxena R                               & Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         & 23300278 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 2.0E-34  & Type 2 diabetes                  & Sladek R                               & A genome-wide association study identifies novel risk loci for type 2 diabetes.                                                                      & 17293876 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 1.0E-30  & Type 2 diabetes and other traits & Rung J                                 & Genetic variant near IRS1 is associated with type 2 diabetes, insulin resistance and hyperinsulinemia.                                               & 19734900 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 9.0E-30  & Type 2 diabetes                  & Timpson NJ                             & Adiposity-related heterogeneity in patterns of type 2 diabetes susceptibility observed in genome-wide association data.                              & 19056611 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 3.0E-23  & Type 2 diabetes                  & Zeggini E                              & Meta-analysis of genome-wide association data and large-scale replication identifies additional susceptibility loci for type 2 diabetes.             & 18372903 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 3.0E-23  & Type 2 diabetes                  & Zhao W                                 & Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                           & 28869590 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 6.0E-22  & Type 2 diabetes                  & Saxena R                               & Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         & 23300278 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 4.0E-21  & Type 2 diabetes                  & Perry JR                               & Stratifying type 2 diabetes cases by BMI identifies genetic risk variants in LAMA1 and enrichment for risk variants in lean compared to obese cases. & 22693455 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 3.0E-19  & Type 2 diabetes                  & Saxena R                               & Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         & 23300278 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 6.0E-16  & Type 2 diabetes                  & Timpson NJ                             & Adiposity-related heterogeneity in patterns of type 2 diabetes susceptibility observed in genome-wide association data.                              & 19056611 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 2.0E-15  & Type 2 diabetes                  & Hara K                                 & Genome-wide association study identifies three novel loci for type 2 diabetes.                                                                       & 23945395 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 2.0E-15  & Type 2 diabetes                  & Kho AN                                 & Use of diverse electronic medical record systems to identify genetic risk for type 2 diabetes within a genome-wide association study.                & 22101970 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 1.0E-14  & Type 2 diabetes                  & Williams AL                            & Sequence variants in SLC16A11 are a common risk factor for type 2 diabetes in Mexico.                                                                & 24390345 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 8.0E-12  & Type 2 diabetes                  & Takeuchi F                             & Confirmation of multiple risk Loci and genetic impacts by a genome-wide association study of type 2 diabetes in the Japanese population.             & 19401414 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 1.0E-11  & Type 2 diabetes                  & Qi Q                                   & Genetics of Type 2 Diabetes in U.S. Hispanic/Latino Individuals: Results From the Hispanic Community Health Study/Study of Latinos (HCHS/SOL).       & 28254843 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 2.0E-10  & Type 2 diabetes                  & Steinthorsdottir V                     & A variant in CDKAL1 influences insulin response and risk of type 2 diabetes.                                                                         & 17460697 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 1.0E-8   & Type 2 diabetes                  & Scott LJ                               & A genome-wide association study of type 2 diabetes in Finns detects multiple susceptibility variants.                                                & 17463248 & H. sapiens\\\\\n",
       "\t 6934 & TCF7L2 & transcription factor 7 like 2     & rs7903146   & 5.0E-8   & Type 2 diabetes                  & Salonen JT                             & Type 2 diabetes whole-genome association study in four populations: the DiaGen consortium.                                                           & 17668382 & H. sapiens\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 46 × 10\n",
       "\n",
       "| GWAS.results.associatedGenes.primaryIdentifier &lt;chr&gt; | GWAS.results.associatedGenes.symbol &lt;chr&gt; | GWAS.results.associatedGenes.name &lt;chr&gt; | GWAS.results.SNP.primaryIdentifier &lt;chr&gt; | GWAS.results.pValue &lt;chr&gt; | GWAS.results.phenotype &lt;chr&gt; | GWAS.firstAuthor &lt;chr&gt; | GWAS.name &lt;chr&gt; | GWAS.publication.pubMedId &lt;chr&gt; | GWAS.results.associatedGenes.organism.shortName &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 3172 | HNF4A  | hepatocyte nuclear factor 4 alpha | rs4812829   | 3.0E-10  | Type 2 diabetes                  | Kooner JS                              | Genome-wide association study in individuals of South Asian ancestry identifies six new type 2 diabetes susceptibility loci.                         | 21874001 | H. sapiens |\n",
       "| 3172 | HNF4A  | hepatocyte nuclear factor 4 alpha | rs4812829   | 4.0E-10  | Type 2 diabetes                  | Zhao W                                 | Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                           | 28869590 | H. sapiens |\n",
       "| 3172 | HNF4A  | hepatocyte nuclear factor 4 alpha | rs4812829   | 5.0E-8   | Type 2 diabetes                  | Mahajan A                              | Genome-wide trans-ancestry meta-analysis provides insight into the genetic architecture of type 2 diabetes susceptibility.                           | 24509480 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs117229942 | 4.0E-11  | Type 2 diabetes                  | Xue A                                  | Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                  | 30054458 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs34872471  | 1.0E-94  | Type 2 diabetes                  | Bonas-Guarch S                         | Re-analysis of public genetic data reveals a rare X-chromosomal variant associated with type 2 diabetes.                                             | 29358691 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs34872471  | 6.0E-53  | Type 2 diabetes                  | Cook JP                                | Multi-ethnic genome-wide association study identifies novel locus for type 2 diabetes susceptibility.                                                | 27189021 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs34872471  | 3.0E-23  | Type 2 diabetes                  | Imamura M                              | Genome-wide association studies in the Japanese population identify seven novel loci for type 2 diabetes.                                            | 26818947 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs34872471  | 8.0E-8   | Type 2 diabetes                  | Ghassibe-Sabbagh M                     | T2DM GWAS in the Lebanese population confirms the role of TCF7L2 and CDKAL1 in disease susceptibility.                                               | 25483131 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs4506565   | 5.0E-12  | Type 2 diabetes                  | Wellcome Trust Case Control Consortium | Genome-wide association study of 14,000 cases of seven common diseases and 3,000 shared controls.                                                    | 17554300 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs4918796   | 4.0E-13  | Type 2 diabetes                  | Xue A                                  | Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                  | 30054458 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7901695   | 1.0E-48  | Type 2 diabetes                  | Zeggini E                              | Replication of genome-wide association signals in UK samples reveals risk loci for type 2 diabetes.                                                  | 17463249 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7901695   | 1.0E-6   | Type 2 diabetes                  | Lettre G                               | Genome-wide association study of coronary heart disease and its risk factors in 8,090 African Americans: the NHLBI CARe Project.                     | 21347282 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 0.0      | Type 2 diabetes                  | Xue A                                  | Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                  | 30054458 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 1.0E-219 | Type 2 diabetes                  | Zhao W                                 | Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                           | 28869590 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 2.0E-184 | Type 2 diabetes                  | Zhao W                                 | Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                           | 28869590 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 1.0E-139 | Type 2 diabetes                  | Morris AP                              | Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                             | 22885922 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 4.0E-94  | Type 2 diabetes                  | Ng MC                                  | Meta-analysis of genome-wide association studies in African Americans provides insights into the genetic architecture of type 2 diabetes.            | 25102180 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 6.0E-92  | Type 2 diabetes                  | Morris AP                              | Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                             | 22885922 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 8.0E-75  | Type 2 diabetes                  | Mahajan A                              | Genome-wide trans-ancestry meta-analysis provides insight into the genetic architecture of type 2 diabetes susceptibility.                           | 24509480 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 9.0E-75  | Type 2 diabetes                  | Saxena R                               | Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         | 23300278 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 4.0E-73  | Type 2 diabetes                  | Morris AP                              | Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                             | 22885922 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 2.0E-51  | Type 2 diabetes                  | Voight BF                              | Twelve type 2 diabetes susceptibility loci identified through large-scale association analysis.                                                      | 20581827 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 1.0E-48  | Type 2 diabetes                  | Saxena R                               | Genome-wide association analysis identifies loci for type 2 diabetes and triglyceride levels.                                                        | 17463246 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 7.0E-45  | Type 2 diabetes                  | Wood AR                                | Variants in the FTO and CDKAL1 loci have recessive effects on risk of obesity and type 2 diabetes, respectively.                                     | 26961502 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 5.0E-44  | Type 2 diabetes                  | Ng MC                                  | Meta-analysis of genome-wide association studies in African Americans provides insights into the genetic architecture of type 2 diabetes.            | 25102180 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 2.0E-40  | Type 2 diabetes                  | Perry JR                               | Stratifying type 2 diabetes cases by BMI identifies genetic risk variants in LAMA1 and enrichment for risk variants in lean compared to obese cases. | 22693455 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 2.0E-38  | Type 2 diabetes                  | Saxena R                               | Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         | 23300278 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 1.0E-35  | Type 2 diabetes                  | Tabassum R                             | Genome-wide association study for type 2 diabetes in Indians identifies a new susceptibility locus at 2q21.                                          | 23209189 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 3.0E-35  | Type 2 diabetes                  | Saxena R                               | Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         | 23300278 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 2.0E-34  | Type 2 diabetes                  | Sladek R                               | A genome-wide association study identifies novel risk loci for type 2 diabetes.                                                                      | 17293876 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 1.0E-30  | Type 2 diabetes and other traits | Rung J                                 | Genetic variant near IRS1 is associated with type 2 diabetes, insulin resistance and hyperinsulinemia.                                               | 19734900 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 9.0E-30  | Type 2 diabetes                  | Timpson NJ                             | Adiposity-related heterogeneity in patterns of type 2 diabetes susceptibility observed in genome-wide association data.                              | 19056611 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 3.0E-23  | Type 2 diabetes                  | Zeggini E                              | Meta-analysis of genome-wide association data and large-scale replication identifies additional susceptibility loci for type 2 diabetes.             | 18372903 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 3.0E-23  | Type 2 diabetes                  | Zhao W                                 | Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                           | 28869590 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 6.0E-22  | Type 2 diabetes                  | Saxena R                               | Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         | 23300278 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 4.0E-21  | Type 2 diabetes                  | Perry JR                               | Stratifying type 2 diabetes cases by BMI identifies genetic risk variants in LAMA1 and enrichment for risk variants in lean compared to obese cases. | 22693455 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 3.0E-19  | Type 2 diabetes                  | Saxena R                               | Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.         | 23300278 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 6.0E-16  | Type 2 diabetes                  | Timpson NJ                             | Adiposity-related heterogeneity in patterns of type 2 diabetes susceptibility observed in genome-wide association data.                              | 19056611 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 2.0E-15  | Type 2 diabetes                  | Hara K                                 | Genome-wide association study identifies three novel loci for type 2 diabetes.                                                                       | 23945395 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 2.0E-15  | Type 2 diabetes                  | Kho AN                                 | Use of diverse electronic medical record systems to identify genetic risk for type 2 diabetes within a genome-wide association study.                | 22101970 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 1.0E-14  | Type 2 diabetes                  | Williams AL                            | Sequence variants in SLC16A11 are a common risk factor for type 2 diabetes in Mexico.                                                                | 24390345 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 8.0E-12  | Type 2 diabetes                  | Takeuchi F                             | Confirmation of multiple risk Loci and genetic impacts by a genome-wide association study of type 2 diabetes in the Japanese population.             | 19401414 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 1.0E-11  | Type 2 diabetes                  | Qi Q                                   | Genetics of Type 2 Diabetes in U.S. Hispanic/Latino Individuals: Results From the Hispanic Community Health Study/Study of Latinos (HCHS/SOL).       | 28254843 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 2.0E-10  | Type 2 diabetes                  | Steinthorsdottir V                     | A variant in CDKAL1 influences insulin response and risk of type 2 diabetes.                                                                         | 17460697 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 1.0E-8   | Type 2 diabetes                  | Scott LJ                               | A genome-wide association study of type 2 diabetes in Finns detects multiple susceptibility variants.                                                | 17463248 | H. sapiens |\n",
       "| 6934 | TCF7L2 | transcription factor 7 like 2     | rs7903146   | 5.0E-8   | Type 2 diabetes                  | Salonen JT                             | Type 2 diabetes whole-genome association study in four populations: the DiaGen consortium.                                                           | 17668382 | H. sapiens |\n",
       "\n"
      ],
      "text/plain": [
       "   GWAS.results.associatedGenes.primaryIdentifier\n",
       "1  3172                                          \n",
       "2  3172                                          \n",
       "3  3172                                          \n",
       "4  6934                                          \n",
       "5  6934                                          \n",
       "6  6934                                          \n",
       "7  6934                                          \n",
       "8  6934                                          \n",
       "9  6934                                          \n",
       "10 6934                                          \n",
       "11 6934                                          \n",
       "12 6934                                          \n",
       "13 6934                                          \n",
       "14 6934                                          \n",
       "15 6934                                          \n",
       "16 6934                                          \n",
       "17 6934                                          \n",
       "18 6934                                          \n",
       "19 6934                                          \n",
       "20 6934                                          \n",
       "21 6934                                          \n",
       "22 6934                                          \n",
       "23 6934                                          \n",
       "24 6934                                          \n",
       "25 6934                                          \n",
       "26 6934                                          \n",
       "27 6934                                          \n",
       "28 6934                                          \n",
       "29 6934                                          \n",
       "30 6934                                          \n",
       "31 6934                                          \n",
       "32 6934                                          \n",
       "33 6934                                          \n",
       "34 6934                                          \n",
       "35 6934                                          \n",
       "36 6934                                          \n",
       "37 6934                                          \n",
       "38 6934                                          \n",
       "39 6934                                          \n",
       "40 6934                                          \n",
       "41 6934                                          \n",
       "42 6934                                          \n",
       "43 6934                                          \n",
       "44 6934                                          \n",
       "45 6934                                          \n",
       "46 6934                                          \n",
       "   GWAS.results.associatedGenes.symbol GWAS.results.associatedGenes.name\n",
       "1  HNF4A                               hepatocyte nuclear factor 4 alpha\n",
       "2  HNF4A                               hepatocyte nuclear factor 4 alpha\n",
       "3  HNF4A                               hepatocyte nuclear factor 4 alpha\n",
       "4  TCF7L2                              transcription factor 7 like 2    \n",
       "5  TCF7L2                              transcription factor 7 like 2    \n",
       "6  TCF7L2                              transcription factor 7 like 2    \n",
       "7  TCF7L2                              transcription factor 7 like 2    \n",
       "8  TCF7L2                              transcription factor 7 like 2    \n",
       "9  TCF7L2                              transcription factor 7 like 2    \n",
       "10 TCF7L2                              transcription factor 7 like 2    \n",
       "11 TCF7L2                              transcription factor 7 like 2    \n",
       "12 TCF7L2                              transcription factor 7 like 2    \n",
       "13 TCF7L2                              transcription factor 7 like 2    \n",
       "14 TCF7L2                              transcription factor 7 like 2    \n",
       "15 TCF7L2                              transcription factor 7 like 2    \n",
       "16 TCF7L2                              transcription factor 7 like 2    \n",
       "17 TCF7L2                              transcription factor 7 like 2    \n",
       "18 TCF7L2                              transcription factor 7 like 2    \n",
       "19 TCF7L2                              transcription factor 7 like 2    \n",
       "20 TCF7L2                              transcription factor 7 like 2    \n",
       "21 TCF7L2                              transcription factor 7 like 2    \n",
       "22 TCF7L2                              transcription factor 7 like 2    \n",
       "23 TCF7L2                              transcription factor 7 like 2    \n",
       "24 TCF7L2                              transcription factor 7 like 2    \n",
       "25 TCF7L2                              transcription factor 7 like 2    \n",
       "26 TCF7L2                              transcription factor 7 like 2    \n",
       "27 TCF7L2                              transcription factor 7 like 2    \n",
       "28 TCF7L2                              transcription factor 7 like 2    \n",
       "29 TCF7L2                              transcription factor 7 like 2    \n",
       "30 TCF7L2                              transcription factor 7 like 2    \n",
       "31 TCF7L2                              transcription factor 7 like 2    \n",
       "32 TCF7L2                              transcription factor 7 like 2    \n",
       "33 TCF7L2                              transcription factor 7 like 2    \n",
       "34 TCF7L2                              transcription factor 7 like 2    \n",
       "35 TCF7L2                              transcription factor 7 like 2    \n",
       "36 TCF7L2                              transcription factor 7 like 2    \n",
       "37 TCF7L2                              transcription factor 7 like 2    \n",
       "38 TCF7L2                              transcription factor 7 like 2    \n",
       "39 TCF7L2                              transcription factor 7 like 2    \n",
       "40 TCF7L2                              transcription factor 7 like 2    \n",
       "41 TCF7L2                              transcription factor 7 like 2    \n",
       "42 TCF7L2                              transcription factor 7 like 2    \n",
       "43 TCF7L2                              transcription factor 7 like 2    \n",
       "44 TCF7L2                              transcription factor 7 like 2    \n",
       "45 TCF7L2                              transcription factor 7 like 2    \n",
       "46 TCF7L2                              transcription factor 7 like 2    \n",
       "   GWAS.results.SNP.primaryIdentifier GWAS.results.pValue\n",
       "1  rs4812829                          3.0E-10            \n",
       "2  rs4812829                          4.0E-10            \n",
       "3  rs4812829                          5.0E-8             \n",
       "4  rs117229942                        4.0E-11            \n",
       "5  rs34872471                         1.0E-94            \n",
       "6  rs34872471                         6.0E-53            \n",
       "7  rs34872471                         3.0E-23            \n",
       "8  rs34872471                         8.0E-8             \n",
       "9  rs4506565                          5.0E-12            \n",
       "10 rs4918796                          4.0E-13            \n",
       "11 rs7901695                          1.0E-48            \n",
       "12 rs7901695                          1.0E-6             \n",
       "13 rs7903146                          0.0                \n",
       "14 rs7903146                          1.0E-219           \n",
       "15 rs7903146                          2.0E-184           \n",
       "16 rs7903146                          1.0E-139           \n",
       "17 rs7903146                          4.0E-94            \n",
       "18 rs7903146                          6.0E-92            \n",
       "19 rs7903146                          8.0E-75            \n",
       "20 rs7903146                          9.0E-75            \n",
       "21 rs7903146                          4.0E-73            \n",
       "22 rs7903146                          2.0E-51            \n",
       "23 rs7903146                          1.0E-48            \n",
       "24 rs7903146                          7.0E-45            \n",
       "25 rs7903146                          5.0E-44            \n",
       "26 rs7903146                          2.0E-40            \n",
       "27 rs7903146                          2.0E-38            \n",
       "28 rs7903146                          1.0E-35            \n",
       "29 rs7903146                          3.0E-35            \n",
       "30 rs7903146                          2.0E-34            \n",
       "31 rs7903146                          1.0E-30            \n",
       "32 rs7903146                          9.0E-30            \n",
       "33 rs7903146                          3.0E-23            \n",
       "34 rs7903146                          3.0E-23            \n",
       "35 rs7903146                          6.0E-22            \n",
       "36 rs7903146                          4.0E-21            \n",
       "37 rs7903146                          3.0E-19            \n",
       "38 rs7903146                          6.0E-16            \n",
       "39 rs7903146                          2.0E-15            \n",
       "40 rs7903146                          2.0E-15            \n",
       "41 rs7903146                          1.0E-14            \n",
       "42 rs7903146                          8.0E-12            \n",
       "43 rs7903146                          1.0E-11            \n",
       "44 rs7903146                          2.0E-10            \n",
       "45 rs7903146                          1.0E-8             \n",
       "46 rs7903146                          5.0E-8             \n",
       "   GWAS.results.phenotype           GWAS.firstAuthor                      \n",
       "1  Type 2 diabetes                  Kooner JS                             \n",
       "2  Type 2 diabetes                  Zhao W                                \n",
       "3  Type 2 diabetes                  Mahajan A                             \n",
       "4  Type 2 diabetes                  Xue A                                 \n",
       "5  Type 2 diabetes                  Bonas-Guarch S                        \n",
       "6  Type 2 diabetes                  Cook JP                               \n",
       "7  Type 2 diabetes                  Imamura M                             \n",
       "8  Type 2 diabetes                  Ghassibe-Sabbagh M                    \n",
       "9  Type 2 diabetes                  Wellcome Trust Case Control Consortium\n",
       "10 Type 2 diabetes                  Xue A                                 \n",
       "11 Type 2 diabetes                  Zeggini E                             \n",
       "12 Type 2 diabetes                  Lettre G                              \n",
       "13 Type 2 diabetes                  Xue A                                 \n",
       "14 Type 2 diabetes                  Zhao W                                \n",
       "15 Type 2 diabetes                  Zhao W                                \n",
       "16 Type 2 diabetes                  Morris AP                             \n",
       "17 Type 2 diabetes                  Ng MC                                 \n",
       "18 Type 2 diabetes                  Morris AP                             \n",
       "19 Type 2 diabetes                  Mahajan A                             \n",
       "20 Type 2 diabetes                  Saxena R                              \n",
       "21 Type 2 diabetes                  Morris AP                             \n",
       "22 Type 2 diabetes                  Voight BF                             \n",
       "23 Type 2 diabetes                  Saxena R                              \n",
       "24 Type 2 diabetes                  Wood AR                               \n",
       "25 Type 2 diabetes                  Ng MC                                 \n",
       "26 Type 2 diabetes                  Perry JR                              \n",
       "27 Type 2 diabetes                  Saxena R                              \n",
       "28 Type 2 diabetes                  Tabassum R                            \n",
       "29 Type 2 diabetes                  Saxena R                              \n",
       "30 Type 2 diabetes                  Sladek R                              \n",
       "31 Type 2 diabetes and other traits Rung J                                \n",
       "32 Type 2 diabetes                  Timpson NJ                            \n",
       "33 Type 2 diabetes                  Zeggini E                             \n",
       "34 Type 2 diabetes                  Zhao W                                \n",
       "35 Type 2 diabetes                  Saxena R                              \n",
       "36 Type 2 diabetes                  Perry JR                              \n",
       "37 Type 2 diabetes                  Saxena R                              \n",
       "38 Type 2 diabetes                  Timpson NJ                            \n",
       "39 Type 2 diabetes                  Hara K                                \n",
       "40 Type 2 diabetes                  Kho AN                                \n",
       "41 Type 2 diabetes                  Williams AL                           \n",
       "42 Type 2 diabetes                  Takeuchi F                            \n",
       "43 Type 2 diabetes                  Qi Q                                  \n",
       "44 Type 2 diabetes                  Steinthorsdottir V                    \n",
       "45 Type 2 diabetes                  Scott LJ                              \n",
       "46 Type 2 diabetes                  Salonen JT                            \n",
       "   GWAS.name                                                                                                                                           \n",
       "1  Genome-wide association study in individuals of South Asian ancestry identifies six new type 2 diabetes susceptibility loci.                        \n",
       "2  Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                          \n",
       "3  Genome-wide trans-ancestry meta-analysis provides insight into the genetic architecture of type 2 diabetes susceptibility.                          \n",
       "4  Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                 \n",
       "5  Re-analysis of public genetic data reveals a rare X-chromosomal variant associated with type 2 diabetes.                                            \n",
       "6  Multi-ethnic genome-wide association study identifies novel locus for type 2 diabetes susceptibility.                                               \n",
       "7  Genome-wide association studies in the Japanese population identify seven novel loci for type 2 diabetes.                                           \n",
       "8  T2DM GWAS in the Lebanese population confirms the role of TCF7L2 and CDKAL1 in disease susceptibility.                                              \n",
       "9  Genome-wide association study of 14,000 cases of seven common diseases and 3,000 shared controls.                                                   \n",
       "10 Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                 \n",
       "11 Replication of genome-wide association signals in UK samples reveals risk loci for type 2 diabetes.                                                 \n",
       "12 Genome-wide association study of coronary heart disease and its risk factors in 8,090 African Americans: the NHLBI CARe Project.                    \n",
       "13 Genome-wide association analyses identify 143 risk variants and putative regulatory mechanisms for type 2 diabetes.                                 \n",
       "14 Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                          \n",
       "15 Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                          \n",
       "16 Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                            \n",
       "17 Meta-analysis of genome-wide association studies in African Americans provides insights into the genetic architecture of type 2 diabetes.           \n",
       "18 Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                            \n",
       "19 Genome-wide trans-ancestry meta-analysis provides insight into the genetic architecture of type 2 diabetes susceptibility.                          \n",
       "20 Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        \n",
       "21 Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes.                            \n",
       "22 Twelve type 2 diabetes susceptibility loci identified through large-scale association analysis.                                                     \n",
       "23 Genome-wide association analysis identifies loci for type 2 diabetes and triglyceride levels.                                                       \n",
       "24 Variants in the FTO and CDKAL1 loci have recessive effects on risk of obesity and type 2 diabetes, respectively.                                    \n",
       "25 Meta-analysis of genome-wide association studies in African Americans provides insights into the genetic architecture of type 2 diabetes.           \n",
       "26 Stratifying type 2 diabetes cases by BMI identifies genetic risk variants in LAMA1 and enrichment for risk variants in lean compared to obese cases.\n",
       "27 Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        \n",
       "28 Genome-wide association study for type 2 diabetes in Indians identifies a new susceptibility locus at 2q21.                                         \n",
       "29 Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        \n",
       "30 A genome-wide association study identifies novel risk loci for type 2 diabetes.                                                                     \n",
       "31 Genetic variant near IRS1 is associated with type 2 diabetes, insulin resistance and hyperinsulinemia.                                              \n",
       "32 Adiposity-related heterogeneity in patterns of type 2 diabetes susceptibility observed in genome-wide association data.                             \n",
       "33 Meta-analysis of genome-wide association data and large-scale replication identifies additional susceptibility loci for type 2 diabetes.            \n",
       "34 Identification of new susceptibility loci for type 2 diabetes and shared etiological pathways with coronary heart disease.                          \n",
       "35 Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        \n",
       "36 Stratifying type 2 diabetes cases by BMI identifies genetic risk variants in LAMA1 and enrichment for risk variants in lean compared to obese cases.\n",
       "37 Genome-wide association study identifies a novel locus contributing to type 2 diabetes susceptibility in Sikhs of Punjabi origin from India.        \n",
       "38 Adiposity-related heterogeneity in patterns of type 2 diabetes susceptibility observed in genome-wide association data.                             \n",
       "39 Genome-wide association study identifies three novel loci for type 2 diabetes.                                                                      \n",
       "40 Use of diverse electronic medical record systems to identify genetic risk for type 2 diabetes within a genome-wide association study.               \n",
       "41 Sequence variants in SLC16A11 are a common risk factor for type 2 diabetes in Mexico.                                                               \n",
       "42 Confirmation of multiple risk Loci and genetic impacts by a genome-wide association study of type 2 diabetes in the Japanese population.            \n",
       "43 Genetics of Type 2 Diabetes in U.S. Hispanic/Latino Individuals: Results From the Hispanic Community Health Study/Study of Latinos (HCHS/SOL).      \n",
       "44 A variant in CDKAL1 influences insulin response and risk of type 2 diabetes.                                                                        \n",
       "45 A genome-wide association study of type 2 diabetes in Finns detects multiple susceptibility variants.                                               \n",
       "46 Type 2 diabetes whole-genome association study in four populations: the DiaGen consortium.                                                          \n",
       "   GWAS.publication.pubMedId GWAS.results.associatedGenes.organism.shortName\n",
       "1  21874001                  H. sapiens                                     \n",
       "2  28869590                  H. sapiens                                     \n",
       "3  24509480                  H. sapiens                                     \n",
       "4  30054458                  H. sapiens                                     \n",
       "5  29358691                  H. sapiens                                     \n",
       "6  27189021                  H. sapiens                                     \n",
       "7  26818947                  H. sapiens                                     \n",
       "8  25483131                  H. sapiens                                     \n",
       "9  17554300                  H. sapiens                                     \n",
       "10 30054458                  H. sapiens                                     \n",
       "11 17463249                  H. sapiens                                     \n",
       "12 21347282                  H. sapiens                                     \n",
       "13 30054458                  H. sapiens                                     \n",
       "14 28869590                  H. sapiens                                     \n",
       "15 28869590                  H. sapiens                                     \n",
       "16 22885922                  H. sapiens                                     \n",
       "17 25102180                  H. sapiens                                     \n",
       "18 22885922                  H. sapiens                                     \n",
       "19 24509480                  H. sapiens                                     \n",
       "20 23300278                  H. sapiens                                     \n",
       "21 22885922                  H. sapiens                                     \n",
       "22 20581827                  H. sapiens                                     \n",
       "23 17463246                  H. sapiens                                     \n",
       "24 26961502                  H. sapiens                                     \n",
       "25 25102180                  H. sapiens                                     \n",
       "26 22693455                  H. sapiens                                     \n",
       "27 23300278                  H. sapiens                                     \n",
       "28 23209189                  H. sapiens                                     \n",
       "29 23300278                  H. sapiens                                     \n",
       "30 17293876                  H. sapiens                                     \n",
       "31 19734900                  H. sapiens                                     \n",
       "32 19056611                  H. sapiens                                     \n",
       "33 18372903                  H. sapiens                                     \n",
       "34 28869590                  H. sapiens                                     \n",
       "35 23300278                  H. sapiens                                     \n",
       "36 22693455                  H. sapiens                                     \n",
       "37 23300278                  H. sapiens                                     \n",
       "38 19056611                  H. sapiens                                     \n",
       "39 23945395                  H. sapiens                                     \n",
       "40 22101970                  H. sapiens                                     \n",
       "41 24390345                  H. sapiens                                     \n",
       "42 19401414                  H. sapiens                                     \n",
       "43 28254843                  H. sapiens                                     \n",
       "44 17460697                  H. sapiens                                     \n",
       "45 17463248                  H. sapiens                                     \n",
       "46 17668382                  H. sapiens                                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "query3GWASResults"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And let's take a look at the unique gene symbols that were returned:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>GWAS.results.associatedGenes.symbol</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>HNF4A </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>TCF7L2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & GWAS.results.associatedGenes.symbol\\\\\n",
       "  & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & HNF4A \\\\\n",
       "\t4 & TCF7L2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 1\n",
       "\n",
       "| <!--/--> | GWAS.results.associatedGenes.symbol &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 1 | HNF4A  |\n",
       "| 4 | TCF7L2 |\n",
       "\n"
      ],
      "text/plain": [
       "  GWAS.results.associatedGenes.symbol\n",
       "1 HNF4A                              \n",
       "4 TCF7L2                             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "GWASIds <- query3GWASResults[\"GWAS.results.associatedGenes.symbol\"]\n",
    "unique(GWASIds)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
